---
title: "Natural selection in contemporary humans is linked to income and substitution effects"
author: "David Hugh-Jones \\thanks{Corresponding author. School of Economics, University of East Anglia, Norwich, UK. Email: D.Hugh-Jones@uea.ac.uk}, Abdel Abdellaoui \\thanks{Department of Psychiatry, Amsterdam UMC, University of Amsterdam, Amsterdam, The Netherlands. Email: a.abdellaoui@amsterdamumc.nl}"
date: "`r format(Sys.Date(), '%d %B %Y')`"
abstract: "\\noindent Natural selection has been documented in contemporary humans, but  little is known about the mechanisms behind it. We test for natural selection through the association between 33 polygenic scores and fertility, across two generations, using data from UK Biobank (N = 409,629 British subjects with European ancestry). Consistently over time, polygenic scores associated with lower (higher) earnings, education and health are selected for (against). Selection effects are concentrated among lower SES groups, younger parents, people with more lifetime sexual partners, and people not living with a partner. The direction of natural selection is reversed among older parents (22+), or after controlling for age at first live birth. These patterns are in line with economic theories of fertility, in which higher earnings may either increase or decrease fertility via income and substitution effects in the labour market. Studying natural selection can help us understand the genetic architecture of health outcomes: we find evidence in modern day Great Britain for multiple natural selection pressures that vary between subgroups in the direction and strength of their effects, that are strongly related to the socio-economic system, and that may contribute to health inequalities across income groups."
bibliography: "negative-selection.bib"
output: 
  bookdown::pdf_document2:
    toc: false
    latex_engine: xelatex
    number_sections: false
header-includes:
  - \usepackage{subfig}
  - \usepackage{setspace}\doublespacing
  - \usepackage{placeins}
  - \usepackage[format=plain, labelfont={bf,it}, textfont=it]{caption}
  - \usepackage{titlesec}
  - \titleformat*{\section}{\sffamily\LARGE}
  - \titleformat*{\subsection}{\sffamily\itshape\Large}
  - \hypersetup{colorlinks = true, linkcolor = {blue}, linkbordercolor = {white}}
editor_options: 
  markdown: 
    wrap: 72
  chunk_output_type: console
mainfont: Baskerville
mathfont: Baskerville
sansfont: "Gill Sans"
---

```{r setup, include = FALSE}

library(drake)
library(magrittr)
library(dplyr)
library(forcats)
library(ggplot2)
library(tidyr)
library(santoku) # keep this after tidyr as it masks chop
library(purrr)
library(glue)
library(broom)
library(huxtable)

# nasty hack to make add_ashe_income work later:
if (! exists("add_ashe_income")) source("~/import-ukbb-data/import-ukbb-data.R")

drake::loadd(rgs)
drake::loadd(score_names)
drake::loadd(famhist)

options(digits = 2)
options(dplyr.summarise.inform = FALSE) # shut up about summarize
knitr::opts_chunk$set(
        echo = FALSE, 
        warning = FALSE, 
        cache = FALSE, 
        error = TRUE,
        fig.height = 3.5
      )
options(huxtable.long_minus = TRUE)

huxtable::set_default_properties(latex_float = "h!")

theme_set(theme_minimal())
my_hline <- geom_hline(yintercept = 0, colour = "grey20", linetype = "dotted")
my_vline <- geom_vline(xintercept = 0, colour = "grey20", linetype = "dotted")
pval_palette <- c("FALSE" = "grey65", "TRUE" = "#FAC32D")


standard_ggplot <- function (
        dfr, 
        fill_col, 
        n_regs, 
        ...,
        score_col = quo(term), 
        order_idx = 1, 
        n_cats    = NULL
      ) {
  if (! missing(score_col)) score_col <- enquo(score_col)
  mfc <- missing(fill_col)
  fill_col <- if (mfc) quo(NULL) else enquo(fill_col)
  if (missing(n_cats)) {
    n_cats <- if (mfc) 1 else length(unique(pull(dfr, {{fill_col}})))
  }
  
  n_regs <- as.double(n_regs)
  
  dfr %>% mutate(
          !! score_col := pretty_names(!! score_col),
          !! score_col := fct_reorder(!! score_col, estimate, order_abs(order_idx))
        ) %>% 
        rename(p = p.value) %>% 
        ggplot(aes(estimate, {{score_col}}, fill = {{fill_col}}, ...,
          colour = p < 0.05/{{n_regs}})) +
          pval_geom_point() +
          my_vline + 
          pval_scale() +
          my_fill_scale(n = n_cats) +
          my_labs() +
          my_theme()
}


my_theme <- function (...) theme(axis.text.y = element_text(size = 7), ...)


pval_scale <- function (...) scale_color_manual(values = pval_palette, ...)


pval_geom_point <- function (size = 1.5, stroke = 0.5, ...) {
  geom_point(shape = 21, size = size, stroke = stroke, ...)
}


my_fill_scale <- function (n, aesthetics = "fill", ...) {
  if (n == 2) {
    scale_colour_manual(values = c("steelblue4", "#D63C3C"), 
          aesthetics = aesthetics, ...)
    # scale_color_brewer(type = "qual", palette = 3, aesthetics = aesthetics, ...)
  } else {
    scale_color_brewer(aesthetics = aesthetics, ...)
  }
}


my_labs <- function (x = "Effect size", y = "", ...) labs(x = x, y = y, ...)


# for use when reordering factors. Order by the nth score in a group
order_abs <- function (n = 1) {
  function (x) x[n] 
}


comma_num <- function (x) prettyNum(x, big.mark = ",")


pretty_names <- function (names) {
  pretty <- c(
    ADHD_2017               = "ADHD",
    age_at_menarche         = "Age at menarche",
    age_at_menopauze        = "Age at menopause",
    agreeableness           = "Agreeableness",
    ai_substance_use        = "Age at smoking initiation",
    alcohol_schumann        = "Alcohol use",
    alzheimer               = "Alzheimer",
    autism_2017             = "Autism",
    bipolar                 = "Bipolar",
    bmi_combined            = "BMI",
    body_fat                = "Body Fat",
    caffeine                = "Caffeine",
    cannabis                = "Cannabis (ever vs. never)",
    cognitive_ability       = "Cognitive Ability",
    conscientiousness       = "Conscientiousness",
    coronary_artery_disease = "Coronary Artery Disease",
    cpd_substance_use       = "Cigarettes per day",
    diagram_T2D             = "Type 2 Diabetes",
    dpw_substance_use       = "Drinks per week",
    EA2_noUKB               = "Educ. attainment 2 (no UKBB)",
    EA3_excl_23andMe_UK     = "Educ. attainment 3 (no UK)",
    eating_disorder         = "Eating disorder",
    extraversion            = "Extraversion",
    height_combined         = "Height",
    hip_combined            = "Hip circumference",
    MDD_PGC2_noUKB          = "Major Depressive Disorder",
    neuroticism             = "Neuroticism",
    openness                = "Openness",
    sc_substance_use        = "Smoking cessation",
    SCZ2                    = "Schizophrenia",
    si_substance_use        = "Smoking initiation",
    wc_combined             = "Waist circumference",
    whr_combined            = "Waist-hip ratio"
  )
  
  pretty[names]
}


n_in_regs <- function (var, dv = "n_children", data = famhist) {
  fh_subset <- data
  if (dv == "n_children") fh_subset %<>% filter(kids_ss)
  # all PGS have the same NA pattern, so we just use one
  tbl <- table(fh_subset[
                  ! is.na(fh_subset[dv]) & ! is.na(fh_subset$whr_combined), 
                  var
                ])
  
  c(tbl)
}


add_n <- function (var, famhist_var = var, dv = "n_children", data = famhist) {
  var <- as.factor(var)
  n <- n_in_regs(famhist_var, dv, data)
  levels(var) <- sprintf("%s (N = %s)", levels(var), 
                         prettyNum(n[seq_len(nlevels(var))], big.mark = ","))
  # var_n <- sprintf("%s (N = %s)", var, prettyNum(n[var], big.mark = ","))
  # var_n[is.na(var)] <- NA
  var
}


# from https://github.com/dgrtwo/drlib/blob/master/R/reorder_within.R
# for ordering within facets
reorder_within <- function(x, by, within, fun = mean, sep = "___", ...) {
  new_x <- paste(x, within, sep = sep)
  stats::reorder(new_x, by, FUN = fun)
}


scale_x_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_x_discrete(labels = function(x) gsub(reg, "", x), ...)
}

scale_y_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_y_discrete(labels = function(x) gsub(reg, "", x), ...)
}

```

\normalem



```{r NOTES-TODO, eval=FALSE}
"
# Open questions/TODO

* Limit max no. of children in regressions (use an "in_children" bool)
  - not necessary. Very few have abnormally many.
* Consider doing separate regressions for "any children" vs. "n children | 1 or more"
  - heckit too slow and gives crazy answers, just use plm/ols
  - some variables do look different
* If we weight with the GHS, and plot the means of figure 1 using this w
  weighting, most things stay the same - but the change in EA is reversed! EA
  increases over time. Could this be because the sample includes many
  relatively educated older people? If so, then mean effects are *overestimated*
  by figure 1.
* Reweight other regressions (e.g. the omnibus regression in the appendix)?

* If you take "year of last job started" and see if it is after/before
  the first child, you can figure out whether mothers keep working.
  - better still, look at all jobs started and see if any are within e.g. 5 years of
  first child's birth
  - this could then be another way to get at income vs substitution effects. You'd
  predict negative selection effects among mothers who worked. Of course the
  choice to work is endogenous.
* double check which scores are reverse coded!

## LITERATURE NOTES

* Lundberg and Pollak JEP 2007:
    - several models where nonmarital fertility is more likely for poorer women
    - Burdett and Ermisch 2002, Willis 1999, Neal 2004
    - Unmarried parents may not be able to credibly commit to each other
  
* Happel et al. 1984
   - childbirth is chosen to minimize loss of earnings from interrupted work.
   - early childbirth if your skills decay a lot  while childrearing. (No point
     gaining them then losing them.) Late childbirth otherwise.
   - late childbirth if you have acquired skills before marriage.
   - Capital-market imperfections make it wise to delay childbirth (till man has
     high income to cope while wife is at home). More so for men's wages which
     rise faster.
   - empirically: (among married couples) 
     - high-skill occupation women have children later. Higher-earning men have 
     them earlier.
     
- Caucutt et al. 2001
  - high waged and high educated women (of recent cohorts) more likely to
    delay childbirth
    
- Cohen et al. 2013
  - analyse effect of Israeli subsidy
  - along the way, find that income affects fertility negatively at low income,
    positively at high income, supporting Becker and Tomes 76

- Becker and Tomes 1976
  - investment in children quality and quantity when quality must be equal
    across children; and some quality comes from an innate endowment.
  - "observed" quantity-of-children income elasticity rises as income rises
  - while "observed" quality-of-children elasticity falls as income rises, 
    b/c more of total quality must be paid for by parents themselves. 
  - the need to pay more for quality results in lower quantity, since total cost is 
    quantity X quality
  - something fishy about argument somehow...
  
- Lindo 2008
  - men who lose a job have children sooner, but fewer children in total
  - controls for all skills effects
  - so overall effect of income is positive, but timing changes
  
- Monstad et al. 2008
  - using an education reform to vary education
  - no evidence of a causal effect on fertility, but births start later
  * maybe the usual correlation reflects underlying skills. Your data could
    test this.
    
- Autor et al. 2017
  - the China shock (exogenous shock to income) 
  - they use Chinese import penetration by industry (instrumented using
    other countries) times industry employment shares; and separate it for
    males and females
  - shocks to male employment decreased fertility and marriage; 
    increased unmarried mothers
  - shocks to female employment *increase* marriage and motherhood
  * this roughly seems to be the consensus
  * Whereas, you find that male and female *skills* (as 'instrumented' by PGS)
    have the same effect on fertility - i.e. less skills = higher fertility
 
- Baudin et al. 2015
  * look at childlessness vs. fertility (extensive/intensive margin)
  * some people are too poor to afford children
  * 'a husband alleviates the time cost of having children'. But for some groups (see appendix)
    there is a +ve income/fertility gradient
  * fertility decreases with female educn due to subst effect
  * baby boom and bust in US: a decline in childlessness caused by poverty, then an increase
    caused by substn effect ('opportunity cost')
  * fertility decreases with educn both for single mothers and married women
    - slightly (25%) higher decrease per year of educn (Table 2) for single mothers
  
# WHAT MODEL FITS YOUR DATA
## ASSUMPTIONS
* Genetics produce human capital h. 
* h and education are compliments in producing earnings
* There are income and substitution effects on fertility:
  - income effects are stronger at high income levels (Becker and Tomes 1976)
    and among couples
* People seek partners when in the labour market. Women match a person, observe
  their quality, and decide whether to have a first child.
* Then they (may) have more children, either with the partner or not 
  (high quality partners stay around more)
* High quality partners increase household income

## RESULTS
* People with low h are more likely to leave education early and enter the labour market
  - because e and h are complements
  - this is part of the substitution effect, as they expect to spend some time
    raising children rather than being in work
* At low values of h, increases in h reduce fertility (substitution effect dominates)
  at higher values, smaller reductions or increases (income effect stronger)
  - because h predicts income
* In 1-parent households, increases in h reduce fertility (sub. effect), in 
  2-parent households, smaller reductions or increases (income effect stronger)
  - because h predicts income and because stronger income effect among couples
* People with low h start having children earlier (b/c leave educ earlier)
  - so part of the substitution effect comes via this
  - controlling for this the income effect dominates
  - hence increases in h reduce fertility less once we control for AFLB
  
## THOUGHTS
* Should we distinguish between men and women in couples?
  - within couples, male h should increase fertility (via earnings)
  - female h should decrease it (ditto?)
  - see explore-sex-vs-pay.R
  
## RIVAL MECHANISMS
* Lynn 2011 mentions two:
  - social class gradient on ability to delay gratification
  - modern society protects the weak (Darwin 1871, Spencer 1874)
    NB this latter doesn't predict more fertility among "dysgenic" ppl, just
    equal fertility and then genetic drift does its work
    - examples include modern sanitation, post-1800
    - and monogamy!
"
```


Living organisms evolve through natural selection, in which allele frequencies
change in the population through differential reproduction rates. Studying the
mechanisms behind natural selection can help us better understand how individual
differences in complex traits and disease risk arise [@benton2021influence]. Recent work confirms that
natural selection is taking place in modern human populations, using genome-wide
analysis
[@Barban_2016;@beauchamp2016genetic;@kong2017selection;@sanjak2018evidence]. In
particular, genetic variants associated with higher educational attainment are
being selected against, although effect sizes appear small.

As yet we know little about the social mechanisms behind these effects. This
study uses data from UK Biobank [@bycroft2018uk] to learn more. We test for
natural selection on 33 different polygenic scores by estimating their
correlation with fertility. We extend the analysis over two generations, using
data on respondents' number of siblings as well as their number of children.
This is interesting because consistent natural selection over multiple
generations could lead to substantive effects in the long run. Most importantly,
we examine reproductive rates in different subgroups of the population, in order
to uncover patterns that can help illuminate the mechanisms behind modern
natural selection.

We find selection effects on many polygenic scores. Effects are largely consistent
across generations. The strength of natural selection on a polygenic score is
associated with that score's correlation with education and earnings: scores
that predict lower education and earnings are being selected for. Also, across
the board, polygenic scores have stronger relationships with fertility among
specific subgroups. Selection effects are stronger in groups with lower
income and less education, among younger parents, people not living with a partner,
and people with more lifetime sexual partners. Outside these groups, effects are
weaker and often statistically insignificant. In some subgroups, the
direction of selection is even reversed: polygenic scores predicting
higher education and earnings are associated with *higher* fertility.

These patterns are in line with economic theories of fertility
[@becker1960economic]. In these, higher potential earnings have two opposite
effects on fertility: a fertility-increasing *income effect* (higher income
makes children more affordable), and a fertility-lowering *substitution effect*
(time spent on childrearing has a higher cost in foregone earnings). Our
results suggest that the substitution effect dominates for single parents and
younger parents, while among couples and older parents (22+) effects are more
evenly balanced.


# Results

We created polygenic scores for 33 traits in `r comma_num(nrow(famhist))`
individuals, corrected for ancestry using 100 genetic principal components (see
Materials and Methods). Figure \@ref(fig:plot-means-over-time) plots mean
polygenic scores in the sample by 5-year birth intervals. Several scores show
consistent increases or declines over this 30-year period, of the order of 5% of
a standard deviation. These changes could reflect natural selection within the
UK population, but also emigration, or ascertainment bias within the sample.
Respondents have higher income and are better educated than the UK population,
and they may also differ on other unobserved characteristics
[@10.1093/aje/kwx246]. Since richer and more educated people also live longer, this
bias could also increase with age.

```{r plot-means-over-time, fig.height = 6, fig.cap = "Mean polygenic scores (PGS) by birth year in UK Biobank. Points are means for 5-year intervals. Lines are 95% confidence intervals. Green triangles show a significant linear increase over time (p < 0.05/33). Red squares show a significant decrease."}

drake::loadd(pgs_over_time)

pgs_over_time %>% 
      mutate(score_name = pretty_names(score_name)) %>% 
      ggplot(aes(YOB, group = score_name, colour = Change, shape = Change)) + 
      geom_pointrange(aes(ymin = score_lo, y = score, ymax = score_hi), 
        fatten = 1.25) +
      facet_wrap(~score_name, ncol = 5) +
      my_hline +
      scale_color_manual(values = c("-" = "red", "+" = "darkgreen", "o" = "black")) +
      scale_shape_manual(values = c("-" = "square", "+" = "triangle", "o" = "circle open")) +
      theme(
        strip.text  = element_text(size = 6), 
        axis.text.x = element_text(size = 7),
        legend.position = "none"
      ) +
      labs(y = "PGS", x = "Birth year (5 year intervals)")

```


To test directly for natural selection, we regress respondents' number of
children ($y_i$) on their polygenic scores (PGS):


```{=tex}
\begin{equation}
y_i = \alpha + \beta\mathrm{PGS}_i + \varepsilon_i
\label{eq:regression}
\end{equation}
```

```{r calc-sex-ratio}
female_ratio <- mean(famhist$sex == 0)
```

The "selection effect", $\beta$, reflects the strength of natural selection
within the sample. In fact, since polygenic scores are normalized, $\beta$ is
the expected polygenic score among children of the sample
[@beauchamp2016genetic].[^normalization] To learn more about the underlying
mechanisms, we split the sample, starting with basic demographic variables
including education, income and sex. These are all potential sources of
ascertainment bias: as well as the ascertainment for income and education,
mentioned above, the sample sex ratio skews `r female_ratio * 100`% female.

[^normalization]: The selection effect $\beta$ equals $Cov(Y, PGS)/Var(PGS)$
where $Y$ is the number of children. Since PGS are normalized to variance 1,
this reduces to to $Cov(Y, PGS) \equiv E(Y PGS) - E(Y) E(PGS)$, which in turn
reduces to $E(Y PGS)$ as $E(PGS) = 0$. This is the polygenic score weighted by
the number of children, which is the average polygenic score in the next
generation.


```{r plot-income-educ-level, fig.cap = "Selection effects by education and income. Each point represents a single bivariate regression of number of children on a polygenic score.", fig.subcap = c("Age left full-time education", "Household income"), fig.ncol = 1}

drake::loadd(res_edu)
n_regs <- as.double(nrow(res_edu))


res_edu %>% 
      mutate(
        term = pretty_names(term), 
        term = fct_reorder(term, estimate, order_abs(1)),
        "Age left FTE" = fct_relevel(age_fte_cat, "< 16", "16-18", "> 18"),
        "Age left FTE" = add_n(`Age left FTE`, "age_fte_cat")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, colour = p < 0.05/{{n_regs}}, 
        fill = `Age left FTE`)) +
        my_vline +
        pval_geom_point() + 
        pval_scale() +
        my_fill_scale(n = 3, direction = -1) +
        my_labs() +
        my_theme()
   
drake::loadd(res_income)

n_regs <- as.double(nrow(res_income))
res_income %>% 
      mutate(
        term = pretty_names(term), 
        term = fct_reorder(term, estimate, order_abs(1)),
        Income = factor(income_cat, 
          labels = c("< £18K", "£18-30K", "£31-51K", "£52-100K", "> £100K")
        ),
        Income = add_n(Income, "income_cat")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, fill = Income, 
        colour = p < 0.05/{{n_regs}})) +
        my_vline +
        pval_geom_point(alpha = 0.8) + 
        my_fill_scale(n = 5, direction = -1) +
        pval_scale() +
        my_labs() +
        my_theme()
```


Figure \@ref(fig:plot-income-educ-level) plots selection effects for each
polygenic score, grouping respondents by age of completing full-time education,
and by household income. Effects are larger and more significant for the lowest
income category, and for the lowest education category. Note that the overall
effect is not a simple average of the effect among the different subgroups,
because polygenic scores may also shift respondents between the subgroups. For
example, a high polygenic score for educational attainment may predict having
fewer children among early school leavers, but it may also increase the age at
which a respondent leaves school.


```{r calc-income-edu-controlled}

drake::loadd(res_income_controlled, res_edu_controlled)
n_regs_ic <- nrow(res_income_controlled)
n_sig_ic <- sum(res_income_controlled$p.value < 0.05/n_regs_ic)

n_regs_educ <- nrow(res_edu_controlled)
n_sig_educ <- sum(res_edu_controlled$p.value < 0.05/n_regs_educ)
```


These results could be explained by age, if older respondents have lower income
and are less educated, and also show more natural selection on polygenic
scores. However, if we rerun the regressions, interacting the polygenic
score with income category and also with a quadratic in age, the
interaction with income remains significant at 0.05/`r n_regs_ic` for
`r n_sig_ic` out of `r n_regs_ic` regressions. Similarly if we interact
the PGS with age of leaving full time education and a quadratic in age,
the interaction with age leaving full time education remains significant at
0.05/`r n_regs_educ` for `r n_sig_educ` out of `r n_regs_educ` regressions.

Selection effects are also different between men and women (Appendix Figure
\@ref(fig:plot-sexes)). Differences are particularly large for educational
attainment, height and MDD. Several polygenic scores for mental illness and
personality traits are more selected for (or less against) among women,
including major depressive disorder (MDD), schizophrenia and neuroticism, while
extraversion is more selected for among men. Scores for waist circumference and
waist-hip ratio are less selected for among women, and scores for educational
attainment are more selected against, though the difference is only significant
for the ("EA3") educational attainment score of @lee2018gene. One
possible reason for these sex differences is that polygenic scores may affect
fertility via success in marriage markets, and men and women may value different
characteristics in these. 


```{r calc-age-flb-cross}

drake::loadd(res_age_flb_cross)

cor_age_flb <- res_age_flb_cross %>% 
      mutate(category = gsub(":.*", "", term)) %>% 
      pivot_wider(score_name, names_from = category, values_from = estimate) %>%              {cor(.$`age_flb_cat10-22`, .$`age_flb_cat28-52`)}

```


We next focus on variables related to household type and reproductive
strategy. We split males and females by lifetime number of sexual
partners, at the median value of 3 (Figure \@ref(fig:plot-n-partners)a).
For both sexes, selection effects are larger and more significant among
those with more than 3 partners. Next we split respondents by whether
they were living with a spouse or partner at the time of interview.
Effects are larger among those not living with a spouse or partner
(Figure \@ref(fig:plot-n-partners)b). 

Lastly, we split female respondents by age at first live birth
(AFLB).[^aflb-no-men] There is evidence for genetic effects on AFLB
[@Barban_2016], and there is a close link between this variable and number of
children born. Figure \@ref(fig:plot-age-flb)a shows effect sizes estimated
separately for each tercile of AFLB. Several effects are strikingly different
across terciles. ADHD and MDD are selected for amongst the youngest third of
mothers, but selected against among the oldest two-thirds. Educational
attainment is selected for among the oldest two-thirds of mothers, but is not
significantly selected among the youngest third. Similarly, several polygenic
scores for body measurements are selected against only among older mothers. The
correlation between effect sizes for the youngest and oldest terciles is 
`r cor_age_flb`. 

[^aflb-no-men]: This information is unavailable for men.


```{r plot-n-partners, fig.cap = "Selection effects by number of sexual partners and household type.", fig.subcap = c("Lifetime number of sexual partners", "Household type"), fig.ncol = 1, fig.align = "center", fig.height = 4}

sexes_theme <-  my_theme(
                  plot.margin          = unit(c(1, 1, 0, 1), "lines"),
                  legend.position      = "bottom", 
                  legend.box.spacing   = unit(0, "lines"),
                  legend.box.margin    = margin(-5, 0, 0, -85),
                  legend.margin        = margin(0, 0, 0, 0),
                  legend.box.just      = "left",
                  legend.justification = "left",
                  legend.spacing       = unit(1.5, "lines")
                )
sexes_guides <- guides(
                  # fill  = guide_legend(ncol = 1),
                  # color = guide_legend(ncol = 1)
                )

drake::loadd(res_partners)
res_partners %<>%       
      filter(grepl(":", term)) %>% 
      mutate(`N partners` = ifelse(grepl("TRUE:", term), "3 or less", "4 or more")) 


n_regs <- as.double(nrow(res_partners))
res_partners %>% 
      mutate(
        score_name = pretty_names(score_name)
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(
        y      = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        x      = estimate, 
        fill   = `N partners`, 
        colour = p < 0.05/{{n_regs}}
      )) + 
        my_vline +
        pval_geom_point() +
        facet_wrap(vars(sex), scales = "free_y") +
        my_fill_scale(n = 2) +
        pval_scale() +
        sexes_theme +
        my_labs(x = "", y = "") + 
        scale_y_reordered() +
        sexes_guides

drake::loadd(res_with_partner_sex)
res_with_partner_sex %<>% filter(grepl(":", term)) 

n_regs <- as.double(nrow(res_with_partner_sex))

res_with_partner_sex %>% 
      mutate(
        score_name = pretty_names(score_name),
        Household = ifelse(grepl("TRUE", term), "With partner", 
                      "Without partner")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(
        y      = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        x      = estimate, 
        fill   = Household, 
        colour = p < 0.05/{{n_regs}}
      )) + 
        my_vline +
        pval_geom_point() +
        facet_wrap(vars(sex), scales = "free_y") +
        my_fill_scale(n = 2) +
        pval_scale() +
        sexes_theme +
        my_labs(x = "", y = "") + 
        scale_y_reordered() +
        sexes_guides

```


```{r calc-corr-age-flb}

drake::loadd(res_age_flb)
drake::loadd(res_sex)

res_sex_comparison <- res_sex %>% 
      filter(sex == "Female") %>% 
      mutate(score_name = term) %>% 
      select(score_name, term, estimate:conf.high)

res_age_flb %<>% filter(term != "age_flb") 

res_combined <- bind_rows(
        raw      = res_sex_comparison, 
        with_age = res_age_flb %>% select(score_name, term, estimate:conf.high), 
        .id      = "reg.type"
      ) %>% 
      arrange(score_name)

raw_flb_comparison <- res_combined %>% 
      pivot_wider(
        id_cols     = score_name, 
        names_from  = reg.type, 
        values_from = estimate
      )

cor_raw_flb <- cor(raw_flb_comparison$raw, raw_flb_comparison$with_age)
opp_signed <- sum(
        sign(raw_flb_comparison$raw) != sign(raw_flb_comparison$with_age)
      )

```


To investigate this further, we regress *number of children* on polygenic scores
*controlling* for AFLB, again among females (Figure \@ref(fig:plot-age-flb)b).
In `r opp_signed` out of `r length(score_names)` cases, effects change sign when
controls are added. The correlation between effect sizes controlling for AFLB,
and raw effect sizes, is `r cor_raw_flb`. Thus, selection effects seem to come
through two opposing channels: an effect on AFLB, and an opposite-signed effect
on number of children controlling for AFLB. Note that we do not claim that AFLB
is exogenous to an individual's polygenic scores -- indeed, our analyses show
that it is not (Appendix Figure \@ref(fig:plot-age-flb-dv)). Rather, we argue
that AFLB mediates part of the relationship between polygenic scores and
fertility, and that once this is controlled for, the remaining part of the
relationship has the opposite sign.


```{r plot-age-flb, fig.cap = "Selection effects by age at first live birth, and controlling for age at first live birth (women only).", fig.ncol = 1,  fig.subcap = c("Effect sizes by age at first live birth terciles (women only)", "Effect sizes controlling for age at first live birth. Effect sizes without controls (for women only) shown for comparison."), fig.align = "center"}

drake::loadd(res_age_flb_cross)

# this counts each cross term as a separate test:
n_regs <- as.double(nrow(res_age_flb_cross)) 

res_age_flb_cross %>% 
      mutate(
        score_name = pretty_names(score_name),
        score_name = fct_reorder(score_name, estimate, order_abs(1)),
        term       = gsub("age_flb_cat(.*):.*", "\\1", term),
        term       = add_n(term, "age_flb_cat")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(estimate, score_name, fill = term, 
        colour = p < 0.05/{{n_regs}})) +
        my_vline +
        pval_geom_point() +
        pval_scale() +
        my_fill_scale(n = 3, direction = -1, 
          guide = guide_legend(title = "Age at first live birth")) +
        my_labs() +
        my_theme()

n_regs <- as.double(nrow(res_age_flb))

res_combined %>% 
      mutate(
        score_name = pretty_names(score_name),
        score_name = fct_reorder(score_name, estimate, order_abs(2)),
        Regression = ifelse(reg.type == "raw", "Uncontrolled", "Controlled for AFLB")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = score_name, x = estimate, fill = Regression, 
        color = p < 0.05/{{n_regs}})) + 
        my_vline +
        pval_geom_point() +
        pval_scale() +
        my_fill_scale(n = 2) +
        my_labs() +
        my_theme()
```


\FloatBarrier
## Correcting for sampling bias


```{r calc-res-weighted}

drake::loadd(res_wt_flb_weights)
drake::loadd(res_wt_msoa_weights)
drake::loadd(res_wt_age_qual_weights)
drake::loadd(res_sibs_parent_weights)
drake::loadd(res_all)
drake::loadd(res_sex)

res_wt_combined <- bind_rows(
  None                = res_all %>% 
                          filter(
                            reg.type == "controlled", 
                            dep.var == "N children"
                          ) %>% 
                          select(-reg.type, -dep.var, -score_name),
  Geographical        = res_wt_msoa_weights,
  `Age/Qualification` = res_wt_age_qual_weights,
  `Age/Qual/AFLB`     = res_wt_flb_weights,
  .id = "Weights"
)

res_wt_cb_wide <- res_wt_combined %>% 
                    tidyr::pivot_wider(
                      id_cols     = term,
                      names_from  = Weights,
                      values_from = estimate
                    ) %>% 
                    rowwise() %>% 
                    mutate(
                      consistent = all(c_across(-term) > 0) ||
                        all(c_across(-term) < 0)
                    ) %>% 
                    ungroup() %>% 
                    left_join(
                      res_wt_combined %>% 
                        filter(Weights == "None") %>% 
                        select(term, p.value),
                      by = "term"
                    ) %>% 
                    left_join(
                      res_sex %>% 
                        filter(sex == "Female") %>% 
                        select(term, estimate_females = estimate),
                      by = "term"
                    )


rel_sizes <- res_wt_cb_wide %>% 
                filter(
                  consistent, 
                  p.value < 0.05/length(score_names)
                ) %>% 
                mutate(
                  Geographical = Geographical/None,
                  `Age/Qualification` = `Age/Qualification`/None,
                  `Age/Qual/AFLB` = `Age/Qual/AFLB`/estimate_females
                ) %>% 
                select(-consistent, -None, -p.value, -estimate_females) %>% 
                mutate(
                  term = pretty_names(term)
                )

rel_sizes_summary <- rel_sizes %>% 
                summarize(
                  across(-term, ~ c(mean(.x), median(.x)))
                ) %>% 
                mutate(
                  term = c("Mean", "Median")
                )

mean_rel <- c(rel_sizes_summary[1, 1:3])
```


Since UK Biobank participants are not representative of the wider population,
our results suggest that naïve estimates of natural selection are at risk of
ascertainment bias. To correct for this, we weight participants using population
data. We try three alternative weighting schemes: weighting by geography, age
and presence/absence of a partner; weighting by age and highest educational
qualification; and age, highest qualification, and age at first live birth, for
women only. Figure \@ref(fig:plot-res-wt) plots selection effects among the
entire sample, estimated with the three weighting schemes.
Mean effect sizes across all polygenic scores are
increased by a factor of `r mean_rel$Geographical` (geographical weighting), 
`r mean_rel[["Age/Qualification"]]` (age/qualification) or 
`r mean_rel[["Age/Qual/AFLB"]]` (age/qualification/AFLB). Estimates might be
further affected by weighting on other demographic variables. Since the UK Biobank
sample seems to be ascertained in ways that shrink estimates of selection
effects, we suspect that our weighted estimates are still conservative.[^purifying]

[^purifying]: We also check for purifying selection by estimating \@ref(eq:regression) with a quadratic term. See Appendix Figure \@ref(fig:plot-purifying).


```{r plot-res-wt, fig.cap = "Selection effects: weighted regressions. Unweighted estimates are shown for comparison."}

n_regs <- as.double(length(score_names))

res_wt_combined %>% 
  mutate(
    Weights  = fct_relevel(Weights, "Geographical", "Age/Qualification",
                "Age/Qual/AFLB", "None"),
    term     = pretty_names(term),
    term     = fct_reorder(term, estimate, order_abs(1))
  ) %>% 
  rename(p = p.value) %>% 
  ggplot(aes(
    x = estimate, 
    y = term, 
    shape = Weights, 
    fill = Weights, 
    colour = p < 0.05/{{n_regs}}
  )) +
  geom_point(size = 1.5, stroke = 0.5) +
  my_vline +
  pval_scale() +
  scale_fill_manual(values = c(
    "None"              = "grey15", 
    "Geographical"      = "dodgerblue4",
    "Age/Qualification" = "steelblue2",
    "Age/Qual/AFLB"      = "darkseagreen4"
  )) +
  scale_shape_manual(values = c(
    "None" = "square filled", 
    "Geographical" = "circle filled",
    "Age/Qualification" = "circle filled", 
    "Age/Qual/AFLB" = "circle filled"
  )) + 
  labs(x = "Effect size", y = "") +
  my_theme()


```


## Selection in the parents' generation

The UK Biobank data contains information on respondents' number of
siblings. Since respondents' polygenic scores are equal in expectation
to the mean scores of their parents, we can use this to look at selection 
effects in the parents' generation. We estimate equation (\ref{eq:regression})
using *number of siblings* (including the respondent) as the dependent variable.
The parents' generation has an additional source of ascertainment bias: sampling
parents of respondents overweights parents who have many children. For instance,
parents of three children will have, on average, three times more children
represented in UK Biobank than parents of one child. Parents of no children will
by definition not be represented. To compensate, we reweight our preferred
weightings (Age/Qualification) by the inverse of *number of siblings*. 

Selection effects are highly correlated across the two generations, and most
share the same sign (Appendix Figure \@ref(fig:plot-res-sibs)). Absolute effect
size estimates are larger for the parents' generation. We treat this result
cautiously, because when we split respondents up by year of birth, we find few
differences in effect sizes between early- and late-born respondents, for either
generation. In other words, since estimated effect sizes change across
"generations", but do not change over time within either generation, the change
may be due to remaining ascertainment bias within the sample or other effects.
In particular, effect sizes in both generations may depend on polygenic scores'
correlation with childlessness, and we cannot estimate this for the parents'
generation.

Although the direction of selection effects does not change between the
generations, there are other differences. Compared to our standard polygenic
scores (residualized on 100 principal components of genetic data), 
selection effects on unresidualized scores are about ten percent higher
on average for *number of siblings*, whereas effects for *number of children*
barely change (Appendix Figure \@ref(fig:plot-regs-controlled-pcs)). This could
be because earlier fertility is driven more by geographically clustered
deprivation [e.g. via an insurance motive, @rendall1998old], which may correlate
with the broad-scale genetic variation captured by principal components, such as
ancestry.

We also check whether selection effects differ by socio-economic status in the
parents' generation. We have no information about parents' income, so we use the
1971 Townsend deprivation score of respondents' birthplace as a proxy
[@townsend1987deprivation]. Results (Appendix Figure
\@ref(fig:plot-siblings-townsend)) show the same pattern as for respondents:
effect sizes are larger and more often significant in the most deprived areas.


```{r calc-siblings-children}
sib_chn_cor <- cor.test(famhist$n_sibs, famhist$n_children, use = "complete")
stopifnot(sib_chn_cor$p.value < 2e-16)
```


Lastly, the siblings data lets us check for a "quantity-quality tradeoff" between
number of children and number of grandchildren (for the parents' generation). We
do not find any: in fact, the correlation between *number of siblings* and 
*number of children* is positive ($\rho$ = `r sib_chn_cor$estimate`, 
$p < 2 \times 10^{-17}$).


## Economic fertility theory and natural selection

These results show that selection effects are weaker, absent, or even reversed
among some subgroups of the population. One possible explanation for this is
given by the economic theory of fertility 
[@becker1960economic; @willis1973new;@becker1976child]. According to this theory,
increases in a person's wage affect their fertility via two opposing channels.
There is an *income effect* by which children become more affordable, like any
other good. There is also a *substitution effect*: since childrearing has a cost
in time, the opportunity cost of childrearing increases if one's market wage is
higher. The income effect would lead higher earners to have more children. The
substitution effect would lead higher earners to have fewer children. If so,
then genetic variants which affect earnings potential in the labour market may
cause opposing effects on fertility. The income effect will cause natural
selection in favour of earnings-increasing variants. The substitution effect
will do the reverse.

This theory can explain why selection effects have opposite signs when AFLB is
controlled for. Suppose that people are aware of their endowments of human
capital, i.e. skills and characteristics that are valuable in the labour market.
Suppose also that education and innate human capital are complements.
Individuals choose how long to stay in education; after leaving education they
enter the labour market and form families. If so, then people with less human
capital will leave education earlier and have their first child earlier [cf.
@caucutt2002women; @monstad2008education]. This is a pure substitution effect:
the income effect plays no role in the decision to leave education, since people
are not yet earning wages.[^no-borrow] Whilst in the labour market, higher
earners will have more children (i.e. income effects dominate). These
assumptions will lead to the pattern we observe: earnings-increasing variants
will predict later first births (and fewer children overall), but they will also
predict more children within any given age at first live birth.

[^no-borrow]: This argument assumes that capital market imperfections hinder
people from borrowing against future income.

The theory can also explain why the natural selection pressure is weaker among
higher-income parents. @becker1976child show that if parents care about child
quality as well as quantity, under certain assumptions the income effect will be
stronger at higher income levels; this can lead to a U-shaped relationship
between income and fertility. Empirically, @cohen2013financial find that income
decreases fertility at low income levels but increases it at higher income
levels. If, in our data, at high income levels income and substitution effects
are roughly balanced , then income-linked polygenic scores will be neither
selected for nor against.

Lastly, it is often assumed that the substitution effect will dominate for lone
parents, or those in unstable relationships, since they have less opportunity to
share childcare responsibilities, while the income effect will dominate for
couples who are able to reap the gains from specialization [@becker1981treatise]. Indeed, US fertility
decreases faster with education among single mothers than married mothers
[@baudin2015fertility]. This can explain why earnings-increasing genetic variants 
decrease fertility more among single parents than among couples. The same logic 
could explain our results for lifetime number of sexual partners, if this is 
associated with relationship stability.


```{r plot-cors-earnings-educ, fig.cap="Selection effects, with and without controls for age at first live birth, by correlations with earnings and educational attainment. Each point represents one polygenic score.", fig.height = 4}

drake::loadd(res_all)
drake::loadd(res_age_flb)
drake::loadd(ashe_income)

famhist <- add_ashe_income(famhist, ashe_income)

cors_income <- cor(famhist[score_names], famhist$first_job_pay, use = "pairwise")
# TODO: use age_left_edu when this includes college

cors_educ <-   cor(famhist[score_names], famhist$age_fulltime_edu, use = "pairwise")

effect_size <-  res_all %>% 
                  filter(dep.var == "N children", reg.type == "controlled") %>% 
                  pull(estimate)

effect_size_flb <-  res_age_flb %>% 
                      filter(term != "age_flb") %>% 
                      pull(estimate)

dfr <- data.frame(
        PGS          = rep(score_names, 4),
        Correlation  = rep(c("Earnings", "Education"), each = 33, 2),
        cor          = rep(c(cors_income[,1], cors_educ[,1]), 2),
        effect_size  = c(rep(effect_size, 2), rep(effect_size_flb, 2)),
        Controls     = rep(c("None", 
                        "Age at first live birth"), each = 66)
      )

dfr %>% 
      mutate(
        Controls = fct_relevel(Controls, "None"),
        PGS      = ifelse(abs(effect_size) > 0.035, PGS, ""),
        PGS      = sub("(^\\w*?)_.*", "\\1", PGS)
      ) %>% 
      ggplot(aes(effect_size, cor, colour = Correlation)) + 
        geom_point() +
        geom_text(aes(label = PGS), size = 3, colour = "black", 
                    check_overlap = TRUE, nudge_x = 0.005) +
        facet_grid(Correlation ~ Controls, labeller = label_both,
                   scales = "free") + 
        my_vline + 
        my_hline +
        labs(x = "Selection effect", y = "") +
        coord_cartesian(clip = "off") +
        scale_colour_brewer(type = "qual", palette = 2) +
        theme(
          legend.position = "none",
          panel.spacing = unit(20, "pt")
        )

```


## Testing the theory

We test this explanation in two ways. First, the economic theory predicts that
genetic variants will be selected for (or against) in proportion to their effect
on earnings. Figure \@ref(fig:plot-cors-earnings-educ) plots selection effects
on each polygenic score against that score's correlation with earnings in a
respondent's first job, and against its correlation with educational attainment,
a predictor of lifetime earnings. (See also the Supplementary Animations for the
uncorrected effects per AFLB group, which show the correlation reversing after
age 22.) The raw relationships (left column) are strongly negative. If we plot
selection effects controlling for age at first live birth (right column), the
effect reverses and becomes positive. Thus, the labour market appears to play an
important role in natural selection. Substitution effects dominate income
effects overall, which fits the known association between income and lower
fertility [@becker1960economic;@jones2006economic].


```{r calc-res-earn-educ}

drake::loadd(res_ee_control)
drake::loadd(res_all)

res_ee_control %<>% filter(term == score_name)
n_05 <- sum(res_ee_control$p.value < 0.05)
n_05_33 <- sum(res_ee_control$p.value < 0.05/33)

```


Second, we re-estimate equation \@ref(eq:regression) for each polygenic score,
controlling for earnings in first job and education levels. Effect sizes are
generally reduced (Appendix Figure \@ref(fig:plot-res-earn-educ)), and only 
`r n_05_33` out of 33 scores are significant at $p = 0.05/33$. This suggests that
earnings and education indeed mediate the effect of polygenic scores on
fertility.


```{r calc-risk-attitude}

drake::loadd(res_risk_control)
drake::loadd(res_all)

n_risk_sig <- res_risk_control %>% 
            filter(term == 'f.2040.0.0') %>% 
            {sum(.$p.value <= 0.05/33)}

n_pgs_sig <- res_risk_control %>% 
            filter(term == score_name) %>% 
            {sum(.$p.value <= 0.05/33)}

res_compare <- res_all %>% 
  filter(dep.var == "N children", reg.type == "controlled") %>% 
  left_join(res_risk_control %>% filter(term == score_name), by = "score_name")

ratio <- res_compare$estimate.y/res_compare$estimate.x
n_less_sig <- sum(res_compare$p.value.y <= 0.05/33 & res_compare$p.value.x > 0.05/33)
stopifnot(n_less_sig == 0)
```


An alternative theory is that traits selected for are linked to externalizing
behaviour, risk-seeking and low time discount rates, via the channel of early
sexual behaviour [@Mills2020.05.06.081273]. The data here provide some support
for this: scores which might plausibly be linked to externalizing behaviour,
like ADHD and (younger) age at smoking initiation, are selected for. However,
this theory is less good at explaining variation in selection across the full
range of scores, including physical measures, e.g. waist-hip ratio and BMI.
Externalizing behaviour also does not explain why selection effects should work
in the opposite direction among older mothers, whereas the economic theory can
explain this via income effects, as described above. We test the alternative
theory directly by re-estimating equation \@ref(eq:regression) controlling for a
measure of risk attitude (UK Biobank field 2040). While risk attitude is always
a highly significant predictor of number of children, it has little impact on
the effect size or significance of the polygenic scores. The median ratio of
effect sizes between regressions with and without controls is `r median(ratio)`;
all scores which are significant at $p \le 0.05/33$ in uncontrolled regressions
remain so when controlling for risk attitude. Thus, although risk attitude does
predict number of children, it appears not to mediate selection effects.
Overall, we believe that the economic theory is the most likely explanation.


# Discussion

Previous work has documented natural selection in modern populations on variants
underlying polygenic traits [@beauchamp2016genetic; @kong2017selection;
@sanjak2018evidence]. We show that correlations between polygenic scores and
fertility are highly concentrated among specific subgroups of the population,
including people with lower income, lower education, younger first parenthood, 
and more lifetime sexual partners. Indeed, among older mothers (22+), selection effects are
actually reversed. Furthermore, the size of selection effects on a polygenic
score correlates with that score's association with labour market earnings. The
economic theory of fertility provides a parsimonious explanation for this.

Polygenic scores which correlate with high (low) earnings and more (less)
education are being selected against (selected for). In addition, many of the
phenotypes under positive selection are linked to disease risk, or are what many
people would see as undesirable to have. For example, most people would probably
prefer to have high educational attainment, a low risk of ADHD and major
depressive disorder, and a low risk of coronary artery disease, but natural
selection is pushing against genes associated with these traits. Potentially,
this could increase the health burden on modern populations, but that depends on
effect sizes. Our results suggest that naïve estimates can be affected by sample
ascertainment bias. This problem may be less serious in surveys which aim to be
representative (as the UK Biobank does not). However, there is still scope for
bias, since not all respondents consent to the collection of genetic data. For
instance, completion rates for genotype data in the US Health and Retirement
Study were around 80-85% [@hrs2020]. Researchers should be aware of the risks of
ascertainment when studying modern natural selection.

We also do not know how estimated effect sizes of natural selection will change
as more accurate measures of genetic variation are produced. And we are unsure
whether genetic variants underlying other phenotypes will show a similar pattern
of natural selection to those studied here. In addition, genetic effects on
educational attainment have been shown to be inflated in population-based
samples as compared to within-family designs, likely because of indirect genetic 
effects, gene-environment correlations, and/or assortative mating 
[@lee2018gene; @selzam2019comparing; @kong2018nature]. 
In short, it is probably too early to tell whether modern natural selection has a
substantively important effect on the genetic make-up of the population.
Nevertheless, we note that selection effects on our measured polygenic scores
are still relatively small, even after reweighting to account for ascertainment
bias.

Because selection effects are concentrated in lower-income groups, they may also
increase inequality with respect to polygenic scores. For example, Figure
\@ref(fig:plot-mean-EA3-by-income) graphs mean polygenic scores for educational
attainment (EA3) among children from households of different income groups. The
blue bars show the actual means, i.e. parents' mean polygenic score weighted by
number of children. The grey bars show the hypothetical means if all households
had equal numbers of children. Natural selection against genes associated with
educational attainment, which lowers the mean, is stronger at the bottom of the
income distribution, and this increases the differences between groups.
Overall, natural selection increases inequality for 29 out of 33 polygenic
scores (Appendix Table \@ref(tab:tbl-inequality-effects)). Since many polygenic
scores are predictive of disease risk, this increase in inequality could
potentially increase health inequalities. In general, the evolutionary history of
anatomically modern humans is related to disease risk [@benton2021influence];
understanding the role of contemporary natural selection may aid in mapping the
genetic architecture of current health disparities.


```{r plot-mean-EA3-by-income, fig.align = "center", fig.cap = "Mean polygenic score for educational attainment (EA3) of children by household income group. Blue is actual. Grey is hypothetical in the absence of selection effects.", fig.width = 4}
income_EA3 <- famhist %>% 
      filter(! is.na(n_children), ! is.na(income_cat)) %>% 
      mutate(
        `Household income` = factor(income_cat, 
                          labels = c("< £18K", "£18-30K", "£31-51K", "£52-100K", 
                                      "> £100K"))
      ) %>% 
      group_by(`Household income`) %>% 
      summarize(
        `Without selection` = mean(EA3_excl_23andMe_UK, na.rm = TRUE),
        Actual              = weighted.mean(EA3_excl_23andMe_UK, n_children, 
                                              na.rm = TRUE)
      ) %>% 
      tidyr::pivot_longer(-`Household income`) %>% 
      rename(
        `Children's mean EA3` = value
      )

income_EA3 %>% 
              ggplot(aes(`Household income`, `Children's mean EA3`, fill = name)) +
                geom_col(position = "dodge") +
                scale_fill_manual(values = c("navy", "grey60")) +
                theme(
                  legend.background = element_rect(fill = "white", colour = NA),
                  legend.title = element_blank(), 
                  legend.position = c(.25, .85)
                )
```


Existing evidence on human natural selection has led some to "biocosmic
pessimism", including speculation that the Fermi paradox may be resolved by
dysgenic selection among alien species [@sarraf2019modernity]. Others are more
sanguine, and argue that natural selection's effects are outweighed by
environmental improvements, such as those underlying (e.g.) the Flynn effect
[@flynn1987massive]. The evidence here may add some nuance to this debate.
Patterns of natural selection have been relatively consistent across the past two
generations, but they are not the outcome of a universal, society-wide phenomenon.
Instead they result from a sum of opposing forces, operating in different
parts of society and pulling in different directions.

Any model of fertility is implicitly a model of natural selection, but so far,
the economic and human genetics literatures have developed in parallel.
Integrating the two could deepen our understanding of natural selection in
modern societies. Economics possesses a range of theoretical models on the
effects of skills, education and income [see @hotz1997economics;
@lundberg2007american]. One perennial problem is how to test these theories in a
world where education, labour and marriage markets all interact. Genetic data,
such as polygenic scores, could help to pin down the direction of causality,
for example via Mendelian randomization [@davey2003mendelian].
Conversely, theory and empirical results from economics can shine a light on the
mechanisms behind natural selection, and thereby on the nature of individual
differences in complex traits and disease risk.


# Materials and methods

We use participant data from UK Biobank [@bycroft2018uk], which has received
ethical approval from the National Health Service North West Centre for Research
Ethics Committee (reference: 11/NW/0382). We limit the sample to white British
participants of European descent, as defined by genetic estimated ancestry and
self-identified ethnic group [@abdellaoui2019genetic], giving a sample size of
`r comma_num(nrow(famhist))`. For regressions on number of children we use
participants over 45, since most fertility is completed by this age. This gives
a sample size of `r comma_num(nrow(famhist %>% filter(kids_ss)))`.

Polygenic scores were computed by summing the alleles across ~1.3 million 
genetic variants weighted by their effect sizes as estimated in 33 genome-wide 
association studies (GWASs) that excluded UK Biobank. To control for population 
stratification, we corrected the polygenic scores for 100 principal components (PCs). 
To compute polygenic scores and PCs, the same procedures were followed as 
described in @abdellaoui2019genetic.

Earnings in first job are estimated from mean earnings in the 2007 Annual Survey
of Hours and Earnings, using the SOC 2000 job code (Biobank field 22617).

Population data for weighting is taken from the 2011 UK Census and the 2006
General Household Survey (GHS). Weighting for Age/Qualification and 
Age/Qualification/AFLB weights was done using marginal totals from a linear
model, using the `calibrate()` function in
the R "survey" package [@lumley2020survey]. Geographical weighting was done with
iterative post-stratification using the `rake()` function, on Census Middle Layer
Super Output Areas, sex and presence/absence of a partner.


# Acknowledgements

AA is supported by the Foundation Volksbond Rotterdam and by ZonMw grant 849200011 from The Netherlands Organisation for Health Research and Development. This study was conducted using UK Biobank resources under application numbers 40310 and 19127.

\FloatBarrier
\clearpage
# Appendix

## Natural selection by sex

```{r plot-sexes, fig.cap = "Selection effects by sex. Highlighted lines are significant differences. Highlighted points are significantly different from 0."}

drake::loadd(res_sex)
n_regs <- as.double(nrow(res_sex))

res_sex_wide <- res_sex %>% 
      select(term, estimate, sex, diff.p.value) %>% 
      pivot_wider(names_from = "sex", values_from = "estimate") %>% 
      mutate(
        min_est = pmin(Female, Male),
        max_est = pmax(Female, Male),
        term = pretty_names(term),
        term = fct_reorder(term, Female)
      ) 
  

res_sex %>% 
      mutate(
        term = pretty_names(term),
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, fill = sex, 
            colour = p < 0.05/{{n_regs}})) +
        my_vline +
        geom_linerange(data = res_sex_wide, 
          aes(
            y      = term,
            x      = NULL,
            fill   = NULL,
            xmin   = min_est, 
            xmax   = max_est, 
            colour = diff.p.value < 0.05/{{n_regs}}
          ), 
        group = "") +
        pval_geom_point() +
        pval_scale() +
        my_fill_scale(n = 2) +
        my_labs() +
        my_theme()

```


\FloatBarrier
\clearpage
## Weighted regressions

Table \@ref(tab:tbl-res-wt) gives effect sizes as a proportion of the unweighted
effect size, for all polygenic scores which are consistently signed and which
are significantly different from zero in unweighted regressions.


```{r tbl-res-wt}

rel_sizes %>% 
      arrange(desc(Geographical)) %>% 
      bind_rows(rel_sizes_summary) %>% 
      rename(PGS = term) %>% 
      as_hux() %>% 
      insert_row("", "Weighting", fill = "") %>% 
      set_caption(
        "Weighted effect sizes as a proportion of unweighted effect sizes."
      ) %>% 
      merge_cells(1, 2:4) %>% 
      set_align(1, 2, "centre") %>% 
      set_header_rows(1, TRUE) %>% 
      theme_compact() %>% 
      set_number_format(-1, -1, 2) %>% 
      style_headers(bold = TRUE) %>% 
      set_bottom_border(1, 2:4) %>% 
      set_bottom_border(2, everywhere) %>% 
      set_italic(final(2), 1) %>% 
      set_font_size(8) %>% 
      set_width(0.7) %>% # ensures line wrapping below
      set_col_width(c(.34, .22, .22, .22)) %>% 
      add_footnote(paste0(
        "Only consistently-signed and significant (when unweighted) ",
        "estimates are shown. ",
        "Age/Qual/AFLB as a proportion of unweighted regressions ",
        "including females only."),
        top_padding  = 0,
        left_padding = 0
      )
      
```


\FloatBarrier
\clearpage
## Purifying and diversifying selection


We rerun equation \@ref(eq:regression), adding a quadratic term in $PGS_i$,
and using our age/qualification weights. Scores for hip circumference and drinks
per week show significant purifying selection ($p < 0.05/33$, negative
coefficient). Scores for educational attainment (EA2 and EA3) show significant
diversifying selection ($p < 0.05/33$, positive coefficient), which reduces the
strength of selection against educational attainment at very high levels of the 
PGS. Figure
\@ref(fig:plot-purifying) plots predicted number of children against polygenic
score from these regressions. 

```{r plot-purifying, fig.align = "center", fig.cap = "Purifying/diversifying selection: predicted number of children by polygenic score.", fig.height = 5}

drake::loadd(res_quadratic)
sig <- res_quadratic %>% 
         filter(grepl("^2", term, fixed = TRUE)) %>% 
         filter(p.value < 0.05/33) %>% 
         pull(score_name)

stopifnot(sig == c("dpw_substance_use", "EA2_noUKB", "EA3_excl_23andMe_UK",
                     "hip_combined"))

sig_all_terms <- res_quadratic %>% filter(score_name %in% sig)

op <- par(mfrow = c(2,2))
for (score in sig) {
  coef <- sig_all_terms %>% filter(score_name == score) %>% pull(estimate)
  curve(
         coef[1] + coef[2] * x + coef[3] * x^2,
         xlim = c(-4, 4), 
         xlab = pretty_names(score),
         ylab = "Predicted N children"
       )
}
par(op)

```




\FloatBarrier
\clearpage
## Parents' generation


```{r calc-res-sibs}
drake::loadd(res_wt_age_qual_weights)
drake::loadd(res_children_comparison)

cmp <- left_join(res_wt_age_qual_weights, res_children_comparison, by = "term")

cor_inc_vs_excl_childless <- cor(cmp$estimate.x, cmp$estimate.y)
n_smaller <- sum(abs(cmp$estimate.y) < abs(cmp$estimate.x))

pct_diff <- abs(cmp$estimate.y)/abs(cmp$estimate.x) - 1
median_pct_diff <- median(pct_diff)

```


Figure \@ref(fig:plot-res-sibs) shows regressions of *number of siblings*, i.e.
parents' number of children, on polygenic scores. By definition, members of the
parents' generation who had no children cannot be included in this data. For a
clean comparison with the respondents' generation, we rerun regressions on
*number of children* excluding those with no children, and show
results in the figure.

Excluding childless people in the parents' generation could bias our estimates.
To learn about this, we compare effect sizes excluding and including childless
people in the *current* generation. The correlation
between the two sets of effect sizes is `r cor_inc_vs_excl_childless`. So, patterns across
different scores are broadly similar whether the childless are counted or not. 
However, absolute effect sizes are smaller when the
childless are excluded, for `r n_smaller` out of 33 scores; the median
percentage change is `r percent(median_pct_diff)`. The fact that childless
people have such a strong effect on estimates makes it hard to compare total
effect sizes across generations. In particular, since the parents' generation has
a different distribution of numbers of children, childless people may have had 
more or less effect in that generation.


```{r plot-res-sibs, fig.cap = "Selection effects, respondents' parents vs. respondents."}
drake::loadd(res_sibs_parent_weights)
drake::loadd(res_children_comparison)

res_sibs_kids <-  bind_rows(
                    Siblings = res_sibs_parent_weights, 
                    Children = res_children_comparison,
                    .id = "Dep. var."
                  )

standard_ggplot(res_sibs_kids, `Dep. var.`, n_regs = length(score_names))
```



```{r calc-period-regs}

drake::loadd(res_period_parents)
drake::loadd(res_period_children)

res_period <- bind_rows(res_period_parents, res_period_children)

res_period %<>% 
      tidyr::extract(term, c("period", "score_name"),
        regex="year_split(.*):(.*)")
n_tests <- nrow(res_period)/2


summary_period <- res_period %>% 
      select(-std.error, -statistic) %>% 
      pivot_wider(
        names_from  = period, 
        values_from = c(estimate, conf.low, conf.high, p.value)
      ) %>% 
      group_by(score_name, children) %>% 
      mutate(
        significant = diff.p.value < 0.05/n_tests,
        sign_early  = sign(estimate_early),
        sign_late   = sign(estimate_late),
        size_inc    = abs(estimate_early) < abs(estimate_late),
        change_sign = sign_early != sign_late,
        Change      = case_when(
                        ! significant ~ "Insignificant",
                        change_sign   ~ "Change sign",
                        size_inc      ~ "Size increasing",
                        ! size_inc    ~ "Size decreasing"
                      )
      ) %>% 
      select(score_name, children, Change, significant, change_sign, 
        sign_early, sign_late) %>% 
      mutate(children = ifelse(children, "Respondents", "Parents")) %>% 
      pivot_wider(names_from = children, 
        values_from = c(Change, significant, change_sign, sign_early, 
        sign_late)) %>% 
      mutate(overall = case_when(
              ! change_sign_Parents & ! change_sign_Respondents & 
                sign_late_Parents == sign_early_Parents ~ 
                "Consistent",
              TRUE ~ "Changes direction"
            ))


stopifnot(sum(summary_period$Change_Parents == "Insignificant") == 33)
stopifnot(sum(summary_period$Change_Respondents == "Size increasing") == 4)
stopifnot(sum(summary_period$Change_Respondents == "Change sign") == 1)
stopifnot(sum(summary_period$Change_Respondents == "Insignificant") == 28)
```

As an alternative approach, we run regressions interacting polygenic scores with
birth year, median split at 1950 ("early born" versus "late born"). We use both
*number of children* and *number of siblings* as a dependent variable. We weight
using age/qualification cells, and further adjust for selection in the parents'
generation (see above).

Tables \@ref(tab:tbl-siblings-by-period) and
\@ref(tab:tbl-children-by-period) summarize the results. There is no
evidence for changes in selection effects within the parents' generation. In the
respondents' generation, effect sizes were significantly larger in absolute size 
among the later-born for four polygenic scores: cognitive ability, EA2, EA3 and
extraversion. These changes are inconsistent with the intergenerational change, 
where estimated effect sizes were larger among the earlier, parents' generation.
One score, conscientiousness, showed a significant change in sign, from 
negative to positive effects on fertility.

Overall, there is weak evidence for change over time. The clearest results are
that (a) the direction of selection, and (b) the pattern of relative effect sizes
across scores, are broadly consistent over time.


```{r tbl-siblings-by-period}

period_table <- function (tbl, caption) {
  tbl %>% 
      as_hux() %>%
      set_bold(1, everywhere) %>% 
      set_bottom_border(1, everywhere) %>% 
      
      add_footnote(glue("Significance is measured at p < 0.05/{n_tests}"), 
        border = 0.4, font_size = 8) %>% 
      set_contents(1, everywhere, 
        c("Change", "Number of scores")) %>% 
      set_right_border(final(1), 1, 0) %>% 
      set_top_padding(1) %>% 
      set_bottom_padding(1) %>%
      set_left_padding(everywhere, 1, 0) %>% 
      set_right_padding(everywhere, 2, 0) %>% 
      set_caption(caption) %>% 
      set_caption_width(1)
}

summary_period %>% 
      group_by(Change_Parents) %>% 
      count() %>% 
      period_table("Change in selection effects between parents of early and late born respondents (regressions on number of siblings).")
      
```

```{r tbl-children-by-period}
summary_period %>% 
      group_by(Change_Respondents) %>% 
      count() %>% 
      period_table("Change in selection effects between early and late born respondents (regressions on number of children).")
```

Figure \@ref(fig:plot-siblings-townsend) plots effects on *number of siblings* by
Townsend deprivation quintile of birth area.

```{r plot-siblings-townsend, fig.cap = "Selection effects in the parents' generation by Townsend deprivation quintile of birth area.", fig.align = "center"}

drake::loadd(res_townsend_parents)
drake::loadd(famhist_townsend_71)
n_regs <- nrow(res_townsend_parents)

res_townsend_parents %>% 
  mutate(
    `Townsend quintile` = sub("Quin71(\\d):.*", "\\1", term),
    `Townsend quintile` = add_n(`Townsend quintile`, "Quin71", dv = "n_sibs", 
                                  data = famhist_townsend_71)
  ) %>% 
  standard_ggplot(fill_col    = `Townsend quintile`,
                    n_regs    = n_regs, 
                    score_col = score_name, 
                    order_idx = 5
                  )

rm(famhist_townsend_71)
```



\clearpage
\FloatBarrier
## Selection effects on raw polygenic scores


```{r calc-regs-controlled}

drake::loadd(res_all)

res_all_controlled <- res_all %>% 
  select(reg.type, dep.var, term, estimate) %>% 
  pivot_wider(names_from = c(reg.type, dep.var), values_from = estimate) %>% 
  mutate(
    consistent = sign(`raw_N siblings`) == sign(`raw_N children`) &
                 sign(`raw_N siblings`) == sign(`controlled_N siblings`) &
                 sign(`raw_N siblings`) == sign(`controlled_N children`)
  )

n_pgs  <- as.double(nrow(res_all_controlled))
prop_consistent_controlled <- mean(res_all_controlled$consistent)
controlled_smaller_sibs <- abs(res_all_controlled[["controlled_N siblings"]]) < 
  abs(res_all_controlled[["raw_N siblings"]])
controlled_smaller_children <- abs(res_all_controlled[["controlled_N children"]]) < 
  abs(res_all_controlled[["raw_N children"]])
median_prop_sibs <- median(abs(res_all_controlled[["controlled_N siblings"]]) /
    abs(res_all_controlled[["raw_N siblings"]]))
median_prop_children <- median(abs(res_all_controlled[["controlled_N children"]]) /
    abs(res_all_controlled[["raw_N children"]]))

```

Figure \@ref(fig:plot-regs-controlled-pcs) compares selection effects on
polygenic scores residualized for the top 100 principal components of the
genetic data, to selection effects on raw, unresidualized polygenic scores. In
siblings regressions, effect sizes are larger for raw scores  -- sometimes much
larger, as in the case of height. `r sum(controlled_smaller_sibs)` out of 
`r n_pgs` "raw" effect sizes have a larger absolute value than the corresponding
"residualized" effect size. The median proportion between raw and controlled
effect sizes is `r median_prop_sibs`. Among the children regressions, this no
longer holds. Effect sizes are barely affected by controlling for principal
components.

Overall, `r prop_consistent_controlled * 100` per cent of effect sizes
are consistently signed across all four regressions (on children and
siblings, and with and without residualization).

```{r plot-regs-controlled-pcs, fig.cap = "Selection effects using unresidualized polygenic scores on number of siblings/children."}

drake::loadd(res_all)

n_regs <- as.double(nrow(res_all))
res_all %>% 
      mutate(
        term     = pretty_names(term),
        term     = reorder_within(term, estimate, dep.var, order_abs(1)),
        dep.var  = fct_inorder(dep.var),
        reg.type = fct_recode(reg.type, 
                               `Resid. for 100 PCs`  = "controlled", 
                               Unresidualized        = "raw"
                             )
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, fill = reg.type, 
        colour = p < 0.05/{{n_regs}})) +
        my_vline + 
        facet_wrap(vars(dep.var), scales = "free") +
        pval_geom_point(position = position_dodge(0.4), alpha = 0.8) +
        my_labs() +
        pval_scale() +
        my_fill_scale(n = 2, guide = guide_legend(title = "")) +
        my_theme() +
        scale_y_reordered() +
        theme(
          axis.text.x  = element_text(size = 7),
          legend.text  = element_text(size = 8),
          legend.title = element_text(size = 9)
        )

```

```{r}
drake::loadd(res_pcs)
n_regs <- as.double(nrow(res_pcs))
prop_sig <- res_pcs %>% 
      group_by(dep.var) %>% 
      summarize(sig = sum(p.value <= 0.05/nrow(res_pcs))) %>% 
      pull(sig, name = dep.var)
```

To get a further insight into this we regress *n siblings* and *n children*
on individual principal components. As Figure \@ref(fig:plot-pc-regs)
shows, effects are larger and more significant in siblings regressions.
`r prop_sig[["N siblings"]]` principal components significantly
predicted number of siblings, while only `r prop_sig[["N children"]]`
significantly predicted number of children.

```{r plot-pc-regs, fig.cap = "Selection effects of 100 principal components of genetic data. Absolute effect sizes are plotted. Each dot represents one bivariate regression. Points are jittered on the Y axis.", fig.height = 2}

res_pcs %>% 
      mutate(
        estimate = abs(estimate),
        dep.var  = fct_relevel(dep.var, "N children")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(estimate, dep.var, colour = p < 0.05/{{n_regs}})) + 
        geom_jitter(height = 0.15, width = 0, alpha = 0.8) + 
        scale_x_log10(labels = scales::label_number(accuracy = 0.0001)) +
        labs(x = "Effect size (log scale)", y = "") + 
        pval_scale(aesthetics = "colour")
        

```


\clearpage
\FloatBarrier
## Selection controlling for age at first live birth: respondents' parents


```{r calc-age-birth-parents}

drake::loadd(res_age_birth_parents)
drake::loadd(res_all)

res_sibs_unc <- filter(res_all, dep.var == "N siblings", reg.type == "controlled")
compare_abp_unc <- res_age_birth_parents %>% 
      filter(! term %in% c("fath_age_birth", "moth_age_birth")) %>% 
      left_join(res_sibs_unc, by = "term")

cor_unc_fath <- with(
  compare_abp_unc %>% filter(control == "fath_age_birth"), 
  cor(estimate.x, estimate.y))
cor_unc_moth <- with(
  compare_abp_unc %>% filter(control == "moth_age_birth"), 
  cor(estimate.x, estimate.y))
```

Among the parents' generation, we can control for age at first live
birth using the subsets of respondents who reported their mother's or
father's age, and who had no elder siblings. We run regressions on
*number of siblings* on these subsets, controlling for either parent's
age at their birth. Figure \@ref(fig:plot-age-birth-parents) shows the
results. Effect sizes are very similar, whether controlling for father's
or mother's age. As in the respondents' generation, effect sizes
are negatively correlated with the effect sizes from bivariate
regressions without the age at birth control (father's age at birth: 
$\rho$ `r cor_unc_fath`; mother's age at birth: $\rho$ `r cor_unc_moth`).

```{r plot-age-birth-parents, fig.cap = "Selection effects (parents' generation) among eldest siblings, controlling for parents' age at birth."}


n_regs <- as.double(nrow(res_age_birth_parents)/2)

res_age_birth_parents %>% 
    filter(! term %in% c("fath_age_birth", "moth_age_birth")) %>% 
    mutate(
      term    = pretty_names(term),
      term    = fct_reorder(term, estimate, order_abs(2)),
      Control = ifelse(control == "fath_age_birth", 
                  "Father's age at birth", 
                  "Mother's age at birth"
                )
    ) %>% 
    rename(p = p.value) %>% 
    ggplot(aes(y = term, x = estimate, fill = Control, 
      colour = p < 0.05/{{n_regs}})) + 
      pval_geom_point(alpha = 0.8) +  
      my_vline +
      my_fill_scale(2) +
      pval_scale() +
      my_labs() +
      my_theme()

```


\clearpage
\FloatBarrier
## Effects of polygenic scores on age at first live birth


```{r calc-age-flb-dv}

drake::loadd(res_age_flb_dv)
drake::loadd(res_age_birth_parents_dv)

compare_flb_dv <- left_join(
        res_age_flb_dv           %>% select(term, estimate),
        res_age_birth_parents_dv %>% select(term, estimate, dep.var),
        by = "term"
      )

cor_flb_fath <- with(compare_flb_dv %>% filter(dep.var == "fath_age_birth"),
        cor(estimate.x, estimate.y)
      )
cor_flb_moth <- with(compare_flb_dv %>% filter(dep.var == "moth_age_birth"),
        cor(estimate.x, estimate.y)
      )
```

Our results suggest that polygenic scores may directly correlate with age at
first live birth. Figure \@ref(fig:plot-age-flb-dv) plots estimated effect sizes
from bivariate regressions for respondents. Figure
\@ref(fig:plot-age-birth-parents-dv) does the same for their parents, using only
eldest siblings.[^only-eldest] Effect sizes are reasonably large. They are also
highly correlated across generations. Effect sizes of polygenic scores on
father's age at own birth, and on own age at first live birth, have a
correlation of `r cor_flb_fath`; for mother's age and own age it is 
`r cor_flb_moth`.

[^only-eldest]: Parental AFLB can only be calculated for this group.

```{r plot-age-flb-dv, fig.cap = "Effects of polygenic scores on age at first live birth."}

n_regs <- as.double(nrow(res_age_flb_dv))

res_age_flb_dv %>% 
      mutate(
        term = pretty_names(term),
        term = fct_reorder(term, estimate, order_abs(1))
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(estimate, term, colour = p < 0.05 / {{n_regs}})) + 
      my_vline +
      pval_geom_point(fill = "steelblue4") +
      pval_scale() + 
      my_labs() +
      my_theme()

```


```{r plot-age-birth-parents-dv, fig.cap = "Effects of polygenic scores on parents' age at respondent's birth, eldest siblings."}

n_regs <- as.double(nrow(res_age_birth_parents_dv))

res_age_birth_parents_dv %>% 
      mutate(
        term = pretty_names(term),
        term = fct_reorder(term, estimate, order_abs(1)),
        "Dependent variable" = dplyr::recode(dep.var, 
          "fath_age_birth" = "Father's AFLB", 
          "moth_age_birth" = "Mother's AFLB"
        )
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(estimate, term, fill = `Dependent variable`, 
        colour = p < 0.05 / {{n_regs}})) + 
        my_vline +
        pval_geom_point() +
        pval_scale() +
        my_fill_scale(n = 2) +
        my_labs() +
        my_theme()

```


\FloatBarrier
\clearpage
## Controlling for earnings and education


```{r plot-res-earn-educ, fig.cap = "Selection effects controlling for earnings in first job (estimated by mean earnings from ASHE 2007 for the SOC2000 job code) and education (left education before 16, 16-18, or after 18). Raw effects are shown for comparison. Lines are 95% confidence intervals uncorrected for multiple testing."}

drake::loadd(res_ee_control)
drake::loadd(res_all)

res_ee_control %<>% filter(term == score_name)


res_all %<>% 
  filter(term == score_name, reg.type == "controlled", dep.var == "N children") %>% 
  select(-dep.var, -reg.type)

res_compare <- bind_rows(
                 Raw = res_all, 
                 Controlled = res_ee_control, 
                 .id = "Regression"
               )

standard_ggplot(res_compare, fill_col = Regression, n_regs = 33) +
  geom_linerange(
    aes(
      xmin = estimate - 2 * std.error, 
      xmax = estimate + 2 * std.error
    ), 
    alpha = .5
  )
  
```


\FloatBarrier
\clearpage
## Genetic correlations with EA3


```{r plot-rgs-by-effect-size, fig.cap = "Selection effects plotted against genetic correlation with EA3."}

# TODO: ref source of data.

drake::loadd(rgs)
drake::loadd(res_all)

rgs <- left_join(rgs, 
        res_all %>% filter(reg.type == "controlled", dep.var == "N children"), 
        by = c("p2" = "term"))

rgs_cor <- cor.test(~ rg + estimate, 
                      data = rgs %>%
                               filter(
                                 p2 != "EA3_excl_23andMe_UK", 
                                 p2 != "EA2_noUKB"
                               )
                    )
rgs %>% 
      filter(p2 != "EA3_excl_23andMe_UK", p2 != "EA2_noUKB") %>% 
      ggplot(aes(rg, estimate)) + 
        geom_point() +
        my_vline +
        my_hline +
        labs(x = "Genetic correlation with EA3", y = "Effect on number of children")
          
```

Another way to examine the "earnings" theory of natural selection is to compare
selection effects of polygenic scores with their genetic correlation with educational
attainment (EA3). Since EA3 strongly predicts earnings, if earnings drives
differences in fertility, we'd expect a correlation between the two sets of
results. Figure \@ref(fig:plot-rgs-by-effect-size) shows this is so: the 
correlation, after excluding EA2, is `r rgs_cor$estimate`. Genetic
correlations were calculated using LD score regression from GWAS summary
statistics.


\clearpage
\FloatBarrier
## Effects on inequality

Table \@ref(tab:tbl-inequality-effects) estimates differences in children's mean
polygenic scores between the highest and lowest income groups. Column "With
selection" uses respondents' scores weighted by Age/Qualification times number
of children. Column "Without selection" uses scores weighted by
Age/Qualification only, i.e. if all couples had the same number of children.


```{r tbl-inequality-effects}

drake::loadd(res_ineq)

ineq_comp <-  left_join(
                res_ineq %>% filter(income_cat == 1),
                res_ineq %>% filter(income_cat == 5),
                by = "score"
              )

ineq_comp %<>% mutate(
                 diff_children = wtd.y - wtd.x,
                 diff_hypoth   = unw.y - unw.x,
                 pct_change    = 100 * (diff_children - diff_hypoth)/diff_hypoth
               )

ineq_comp %>% 
      select(
        PGS                 = score, 
        `Without selection` = diff_hypoth,
        `With selection`    = diff_children, 
        `% change`          = pct_change
      ) %>% 
      mutate(
        PGS = pretty_names(PGS)
      ) %>% 
      arrange(desc(`% change`)) %>% 
      as_hux() %>% 
      set_align(everywhere, -1, "right") %>% 
      set_number_format(everywhere, 2:3, 3) %>% 
      set_number_format(everywhere, 4, 1) %>% 
      set_caption("Differences in polygenic scores between highest and lowest income group.") %>% 
      theme_compact() %>% 
      set_width(0.75) %>% 
      set_col_width(c(.4, .2, .2, .2)) %>% 
      set_bottom_border(final(1), everywhere) %>% 
      set_position("center")
```

\clearpage
\FloatBarrier
# References
