---
title: "Human capital mediates natural selection in contemporary humans"
author: "David Hugh-Jones \\thanks{Corresponding author. School of Economics, University of East Anglia, Norwich, UK. Email: D.Hugh-Jones@uea.ac.uk}, Abdel Abdellaoui \\thanks{Department of Psychiatry, Amsterdam UMC, University of Amsterdam, Amsterdam, The Netherlands.}"
date: "`r format(Sys.Date(), '%d %B %Y')`"
abstract: "\\noindent Natural selection has been documented in contemporary humans, but  little is known about the mechanisms behind it. We test for natural selection through the association between 33 polygenic scores and fertility, across two generations, using data from UK Biobank (N = 409,629 British subjects with European ancestry). Consistently over time, polygenic scores associated with lower (higher) earnings, education and health are selected for (against). Selection effects are concentrated among lower SES groups, younger parents, people with more lifetime sexual partners, and people not living with a partner. The direction of natural selection is reversed among older parents (22+), or after controlling for age at first live birth. These patterns are in line with economic theories of fertility, in which earnings-increasing human capital may either increase or decrease fertility via income and substitution effects in the labour market. Studying natural selection can help us understand the genetic architecture of health outcomes: we find evidence in modern day Great Britain for multiple natural selection pressures that vary between subgroups in the direction and strength of their effects, that are strongly related to the socio-economic system, and that may contribute to health inequalities across income groups."
bibliography: "negative-selection.bib"
output: 
  bookdown::pdf_document2:
    toc: false
    latex_engine: xelatex
    number_sections: false
header-includes:
  - \usepackage{subfig}
  - \usepackage{setspace}\doublespacing
  - \usepackage{placeins}
  - \usepackage[format=plain, labelfont={bf,it}, textfont=it]{caption}
  - \usepackage{titlesec}
  - \titleformat*{\section}{\sffamily\LARGE}
  - \titleformat*{\subsection}{\sffamily\itshape\Large}
  - \hypersetup{colorlinks = true, linkcolor = {blue}, linkbordercolor = {white}}
  - \usepackage{amsthm}
  - \theoremstyle{plain}
  - \newtheorem{lem}{\protect\lemmaname}
  - \providecommand{\lemmaname}{Lemma}
  - \usepackage{lineno}
  - \linenumbers
editor_options: 
  markdown: 
    wrap: 72
  chunk_output_type: console
mainfont: Baskerville
mathfont: Baskerville
sansfont: "Gill Sans"
---

```{r setup, include = FALSE}

library(drake)
library(magrittr)
library(dplyr)
library(forcats)
library(ggplot2)
library(tidyr)
library(santoku) # keep this after tidyr as it masks chop
library(purrr)
library(glue)
library(broom)
library(huxtable)

# nasty hack to make add_ashe_income work later:
if (! exists("add_ashe_income")) source("~/import-ukbb-data/import-ukbb-data.R")

drake::loadd(rgs)
drake::loadd(score_names)
drake::loadd(famhist)

options(digits = 2)
options(dplyr.summarise.inform = FALSE) # shut up about summarize
knitr::opts_chunk$set(
        echo = FALSE, 
        warning = FALSE, 
        cache = FALSE, 
        error = TRUE,
        fig.height = 3.5
      )
options(huxtable.long_minus = TRUE)

huxtable::set_default_properties(latex_float = "h!")

theme_set(theme_minimal())

my_hline <- geom_hline(yintercept = 0, colour = "grey20", linetype = "dotted")
my_vline <- geom_vline(xintercept = 0, colour = "grey20", linetype = "dotted")
pval_palette <- c("FALSE" = "grey65", "TRUE" = "#FAC32D")


standard_ggplot <- function (
        dfr, 
        fill_col, 
        n_regs, 
        ...,
        score_col = quo(term), 
        order_idx = 1, 
        fill_direction = 1,
        n_cats    = NULL,
        conf_int  = NULL
      ) {
  if (! missing(score_col)) score_col <- enquo(score_col)
  
  mfc <- missing(fill_col)
  fill_col <- if (mfc) quo(NULL) else enquo(fill_col)
  if (missing(n_cats)) {
    n_cats <- if (mfc) 1 else length(unique(pull(dfr, {{fill_col}})))
  }
  
  if (is.null(conf_int)) conf_int <- "conf.low" %in% names(dfr)
  
  n_regs <- as.double(n_regs)
 
  confint_geom_segment <- if (conf_int) confint_geom_segment() else NULL
  
  dfr %>% mutate(
          !! score_col := pretty_names(!! score_col),
          !! score_col := fct_reorder(!! score_col, estimate, order_abs(order_idx))
        ) %>% 
        rename(p = p.value) %>% 
        ggplot(aes(estimate, {{score_col}}, yend = {{score_col}}, 
                     fill = {{fill_col}}, ..., colour = p < 0.05/{{n_regs}})) +
          my_vline + 
          confint_geom_segment +
          pval_geom_point(fill = if(rlang::quo_is_null(fill_col)) "steelblue") +
          pval_scale() +
          my_fill_scale(n = n_cats, direction = fill_direction) +
          my_labs() +
          my_theme() +
          scale_linetype_manual(name = "", values = c("95% c.i." = 1))

}


my_theme <- function (...) theme(axis.text.y = element_text(size = 7), ...)


pval_scale <- function (...) scale_color_manual(values = pval_palette, ...)


pval_geom_point <- function (size = 1.5, stroke = 0.5, fill = NULL, ...) {
  if (is.null(fill)) {
    geom_point(shape = 21, size = size, stroke = stroke, ...)
  } else {
    geom_point(shape = 21, size = size, stroke = stroke, fill = fill, ...)
  }
}


confint_geom_segment <- function () {
  geom_segment(aes(x = conf.low, xend = conf.high, linetype = "95% c.i."), 
                    color = "grey60", alpha = 0.25, size = 1.1, 
                    lineend = "round")
}


my_fill_scale <- function (n, aesthetics = "fill", ...) {
  
  fill_colors <- c("steelblue4", "#D63C3C")
  
  if (n <= 2) {
    dots <- list(...)
    dots$direction <- NULL
    do.call(scale_colour_manual, c(list(values = fill_colors[1:n],
                                        aesthetics = aesthetics), 
                                   dots))
  } else {
    scale_color_brewer(aesthetics = aesthetics, ...)
  }
}


my_labs <- function (x = "Effect size", y = "", ...) labs(x = x, y = y, ...)


# for use when reordering factors. Order by the nth score in a group
order_abs <- function (n = 1) {
  function (x) x[n] 
}


comma_num <- function (x) prettyNum(x, big.mark = ",")


pretty_names <- function (names) {
  pretty <- c(
    ADHD_2017               = "ADHD",
    age_at_menarche         = "Age at menarche",
    age_at_menopauze        = "Age at menopause",
    agreeableness           = "Agreeableness",
    ai_substance_use        = "Age at smoking initiation",
    alcohol_schumann        = "Alcohol use",
    alzheimer               = "Alzheimer",
    autism_2017             = "Autism",
    bipolar                 = "Bipolar",
    bmi_combined            = "BMI",
    body_fat                = "Body Fat",
    caffeine                = "Caffeine",
    cannabis                = "Cannabis (ever vs. never)",
    cognitive_ability       = "Cognitive Ability",
    conscientiousness       = "Conscientiousness",
    coronary_artery_disease = "Coronary Artery Disease",
    cpd_substance_use       = "Cigarettes per day",
    diagram_T2D             = "Type 2 Diabetes",
    dpw_substance_use       = "Drinks per week",
    EA2_noUKB               = "Educ. attainment 2 (no UKBB)",
    EA3_excl_23andMe_UK     = "Educ. attainment 3 (no UK)",
    eating_disorder         = "Eating disorder",
    extraversion            = "Extraversion",
    height_combined         = "Height",
    hip_combined            = "Hip circumference",
    MDD_PGC2_noUKB          = "Major Depressive Disorder",
    neuroticism             = "Neuroticism",
    openness                = "Openness",
    sc_substance_use        = "Smoking cessation",
    SCZ2                    = "Schizophrenia",
    si_substance_use        = "Smoking initiation",
    wc_combined             = "Waist circumference",
    whr_combined            = "Waist-hip ratio"
  )
  
  pretty[names]
}


n_in_regs <- function (var, dv = "n_children", data = famhist) {
  fh_subset <- data
  if (dv == "n_children") fh_subset %<>% filter(kids_ss)
  # all PGS have the same NA pattern, so we just use one
  tbl <- table(fh_subset[
                  ! is.na(fh_subset[dv]) & ! is.na(fh_subset$whr_combined), 
                  var
                ])
  
  c(tbl)
}


add_n <- function (var, famhist_var = var, dv = "n_children", data = famhist) {
  var <- as.factor(var)
  n <- n_in_regs(famhist_var, dv, data)
  levels(var) <- sprintf("%s (N = %s)", levels(var), 
                         prettyNum(n[seq_len(nlevels(var))], big.mark = ","))
  # var_n <- sprintf("%s (N = %s)", var, prettyNum(n[var], big.mark = ","))
  # var_n[is.na(var)] <- NA
  var
}


# from https://github.com/dgrtwo/drlib/blob/master/R/reorder_within.R
# for ordering within facets
reorder_within <- function(x, by, within, fun = mean, sep = "___", ...) {
  new_x <- paste(x, within, sep = sep)
  stats::reorder(new_x, by, FUN = fun)
}


scale_x_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_x_discrete(labels = function(x) gsub(reg, "", x), ...)
}

scale_y_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_y_discrete(labels = function(x) gsub(reg, "", x), ...)
}

```

\normalem



```{r NOTES-TODO, eval=FALSE}
"

# TODO response to PNAS

* Only use education, not first job pay, in "controlled" regressions; check
  description is correct - done.
* Add CIs - done.
* Clarify collider bias/endogeneity - done.
* Formalize theory - done.
  - Comment on results where effects are actually reversed. This works
    nicely if you throw in a money cost - done computational example.
* Clarify causality? - done...
  - Helps to deal with the "collider bias" argument of reviewer
  - Might also be a better test of mediation via human capital
  - One other possibility. If we use only people with first profession
    teacher, that (as it happens) shuts down the EA3-income correlation.
    It also shuts down the EA3-fertility relationship, suggesting that
    income is indeed the mediator.
* Clarify "what we learn about genetic links to fertility by using economic theory to interpret relationships between polygenic scores and fertility."
  - Point is that we can explain how natural selection is occurring.
* "Discuss empirical predictions for AFLB grounded in dynamic theory"
  - hmm, tricky, maybe we can include it in discussion of economic results at
  end of formal section.
  - think our theory with 2 periods probably covers this
* Why don't women see stronger effects? But note, they do: not significant
  after correction for multiple testing, but a 25% difference. 
  - I think we leave this for reviewers.
* Quadratic regressions for parents - DONE

# Open questions/TODO

* Limit max no. of children in regressions (use an "in_children" bool)
  - not necessary. Very few have abnormally many.
* Consider doing separate regressions for "any children" vs. "n children | 1 or more"
  - heckit too slow and gives crazy answers, just use plm/ols
  - some variables do look different
* If we weight with the GHS, and plot the means of figure 1 using this w
  weighting, most things stay the same - but the change in EA is reversed! EA
  increases over time. Could this be because the sample includes many
  relatively educated older people? If so, then mean effects are *overestimated*
  by figure 1.
* Reweight other regressions (e.g. the omnibus regression in the appendix)?

* If you take "year of last job started" and see if it is after/before
  the first child, you can figure out whether mothers keep working.
  - better still, look at all jobs started and see if any are within e.g. 5 years of
  first child's birth
  - this could then be another way to get at income vs substitution effects. You'd
  predict negative selection effects among mothers who worked. Of course the
  choice to work is endogenous.
* double check which scores are reverse coded!
  
# WHAT MODEL FITS YOUR DATA
## ASSUMPTIONS
* Genetics produce human capital h. 
* h and education are compliments in producing earnings
* There are income and substitution effects on fertility:
  - income effects are stronger at high income levels (Becker and Tomes 1976)
    and among couples
* People seek partners when in the labour market. Women match a person, observe
  their quality, and decide whether to have a first child.
* Then they (may) have more children, either with the partner or not 
  (high quality partners stay around more)
* High quality partners increase household income

## RESULTS
* People with low h are more likely to leave education early and enter the labour market
  - because e and h are complements
  - this is part of the substitution effect, as they expect to spend some time
    raising children rather than being in work
* At low values of h, increases in h reduce fertility (substitution effect dominates)
  at higher values, smaller reductions or increases (income effect stronger)
  - because h predicts income
* In 1-parent households, increases in h reduce fertility (sub. effect), in 
  2-parent households, smaller reductions or increases (income effect stronger)
  - because h predicts income and because stronger income effect among couples
* People with low h start having children earlier (b/c leave educ earlier)
  - so part of the substitution effect comes via this
  - controlling for this the income effect dominates
  - hence increases in h reduce fertility less once we control for AFLB
  
## THOUGHTS
* Should we distinguish between men and women in couples?
  - within couples, male h should increase fertility (via earnings)
  - female h should decrease it (ditto?)
  - see explore-sex-vs-pay.R
  
## RIVAL MECHANISMS
* Lynn 2011 mentions two:
  - social class gradient on ability to delay gratification
  - modern society protects the weak (Darwin 1871, Spencer 1874)
    NB this latter doesn't predict more fertility among "dysgenic" ppl, just
    equal fertility and then genetic drift does its work
    - examples include modern sanitation, post-1800
    - and monogamy!
"
```


Living organisms evolve through natural selection, in which allele frequencies
change in the population through differential reproduction rates. Studying the
mechanisms behind natural selection can help us better understand how individual
differences in complex traits and disease risk arise [@benton2021influence].
Recent work confirms that natural selection is taking place in modern human
populations, using genome-wide analysis
[@Barban_2016;@beauchamp2016genetic;@kong2017selection;@sanjak2018evidence]. In
particular, genetic variants associated with higher educational attainment are
being selected against, although effect sizes appear small.

As yet we know little about the social mechanisms behind these effects. This
study uses data from UK Biobank [@bycroft2018uk] to learn more. We test for
natural selection on 33 different polygenic scores by estimating their
correlation with fertility. We extend the analysis over two generations, using
data on respondents' number of siblings as well as their number of children.
This is interesting because consistent natural selection over multiple
generations could lead to substantive effects in the long run. Most importantly,
we examine reproductive rates in different subgroups of the population, in order
to uncover patterns that can help illuminate the mechanisms behind modern
natural selection.

We find selection effects on many polygenic scores. Effects are largely consistent
across generations. The strength of natural selection on a polygenic score is
associated with that score's correlation with education and earnings: scores
that predict lower education and earnings are being selected for. Also, across
the board, polygenic scores have stronger relationships with fertility among
specific subgroups. Selection effects are stronger in groups with lower
income and less education, among younger parents, people not living with a partner,
and people with more lifetime sexual partners. Outside these groups, effects are
weaker and often statistically insignificant. In some subgroups, the
direction of selection is even reversed: polygenic scores predicting
higher education and earnings are associated with *higher* fertility.

These patterns are in line with economic theories of fertility
[@becker1960economic]. In these, higher potential earnings have two opposite
effects on fertility: a fertility-increasing *income effect* (higher income
makes children more affordable), and a fertility-lowering *substitution effect*
(time spent on childrearing has a higher cost in foregone earnings). Our results
suggest that the substitution effect dominates for single parents and younger
parents, while among couples and older parents (22+) effects are more evenly
balanced. Thus, contemporary natural selection on polygenic scores can be
explained by their correlation with earnings-increasing human capital. Below, we 
show that a simple model of human capital, education and fertility choices 
can give rise to our key empirical results.
 

# Results


```{r calc-sex-ratio}
female_ratio <- mean(famhist$sex == 0)
```


We created polygenic scores for 33 traits in `r comma_num(nrow(famhist))`
individuals, corrected for ancestry using 100 genetic principal components (see
Materials and Methods). Figure \@ref(fig:plot-means-over-time) plots mean
polygenic scores in the sample by 5-year birth intervals. Several scores show
consistent increases or declines over this 30-year period, of the order of 5% of
a standard deviation. These changes could reflect natural selection within the
UK population, but also emigration, or ascertainment bias within the sample.
Respondents have higher income and are better educated than the UK population,
and they may also differ on other unobserved characteristics
[@10.1093/aje/kwx246]. Since richer and more educated people also live longer, this
bias could also increase with age. Lastly, the sample sex ratio skews 
`r female_ratio * 100`% female.


```{r plot-means-over-time, fig.height = 6, fig.cap = "Mean polygenic scores (PGS) by birth year in UK Biobank. Points are means for 5-year intervals. Lines are 95% confidence intervals. Green triangles show a significant linear increase over time (p < 0.05/33). Red squares show a significant decrease."}

drake::loadd(pgs_over_time)

pgs_over_time %>% 
      mutate(score_name = pretty_names(score_name)) %>% 
      ggplot(aes(YOB, group = score_name, colour = Change, shape = Change)) + 
      geom_pointrange(aes(ymin = score_lo, y = score, ymax = score_hi), 
        fatten = 1.25) +
      facet_wrap(~score_name, ncol = 5) +
      my_hline +
      scale_color_manual(values = c("-" = "red", "+" = "darkgreen", "o" = "black")) +
      scale_shape_manual(values = c("-" = "square", "+" = "triangle", "o" = "circle open")) +
      theme(
        strip.text  = element_text(size = 6), 
        axis.text.x = element_text(size = 7),
        legend.position = "none"
      ) +
      labs(y = "PGS", x = "Birth year (5 year intervals)")

```


To test more directly for natural selection, we regress respondents' number of
children ($N_i$) on their polygenic scores (PGS):


```{=tex}
\begin{equation}
N_i = \alpha + \beta\mathrm{PGS}_i + \varepsilon_i
\label{eq:regression}
\end{equation}
```


The "selection effect", $\beta$, reflects the strength of natural selection
within the sample. In fact, since polygenic scores are normalized, $\beta$ is
the expected polygenic score among children of the sample
[@beauchamp2016genetic].[^normalization] Note that equation 
\eqref{eq:regression} does not include many possible environmental and
genetic confounds that could affect fertility, and as a result, $\beta$ is not
an estimate of the causal effect of a polygenic score on fertility. However, 
natural selection is a correlational phenomenon, not a causal one: if a 
polygenic score correlates with low fertility, it is being selected against, 
whatever the underlying causal relationship. 

[^normalization]: The selection effect $\beta$ equals $Cov(N, PGS)/Var(PGS)$
where $N$ is the number of children. Since PGS are normalized to variance 1 and
mean 0, this reduces to 
$Cov(N, PGS) \equiv E(N\ PGS) - E(N) E(PGS) = E(N\ PGS)$. This is the polygenic 
score weighted by the number of children, which is the average polygenic score 
in the next generation.


```{r calc-res-weighted}

drake::loadd(res_wt_flb_weights)
drake::loadd(res_wt_msoa_weights)
drake::loadd(res_wt_age_qual_weights)
drake::loadd(res_sibs_parent_weights)
drake::loadd(res_all)
drake::loadd(res_sex)

res_wt_combined <- bind_rows(
  None                = res_all %>% 
                          filter(
                            reg.type == "controlled", 
                            dep.var == "N children"
                          ) %>% 
                          select(-reg.type, -dep.var, -score_name),
  Geographical        = res_wt_msoa_weights,
  `Age/Qualification` = res_wt_age_qual_weights,
  `Age/Qual/AFLB`     = res_wt_flb_weights,
  .id = "Weights"
)

res_significance <- res_wt_combined %>% 
                      group_by(term) %>% 
                      summarize(
                        all = all(p.value < 0.05/33), 
                        any = any(p.value < 0.05/33)
                      )
res_sig_all <- sum(res_significance$all)
res_sig_any <- sum(res_significance$any)

res_wt_cb_wide <- res_wt_combined %>% 
                    tidyr::pivot_wider(
                      id_cols     = term,
                      names_from  = Weights,
                      values_from = estimate
                    ) %>% 
                    rowwise() %>% 
                    mutate(
                      consistent = all(c_across(-term) > 0) ||
                        all(c_across(-term) < 0)
                    ) %>% 
                    ungroup() %>% 
                    left_join(
                      res_wt_combined %>% 
                        filter(Weights == "None") %>% 
                        select(term, p.value),
                      by = "term"
                    ) %>% 
                    left_join(
                      res_sex %>% 
                        filter(sex == "Female") %>% 
                        select(term, estimate_females = estimate),
                      by = "term"
                    )


rel_sizes <- res_wt_cb_wide %>% 
                filter(
                  consistent, 
                  p.value < 0.05/length(score_names)
                ) %>% 
                mutate(
                  Geographical = Geographical/None,
                  `Age/Qualification` = `Age/Qualification`/None,
                  `Age/Qual/AFLB` = `Age/Qual/AFLB`/estimate_females
                ) %>% 
                select(-consistent, -None, -p.value, -estimate_females) %>% 
                mutate(
                  term = pretty_names(term)
                )

rel_sizes_summary <- rel_sizes %>% 
                summarize(
                  across(-term, ~ c(mean(.x), median(.x)))
                ) %>% 
                mutate(
                  term = c("Mean", "Median")
                )

mean_rel <- c(rel_sizes_summary[1, 1:3])
```


To correct for ascertainment bias, we weight participants using population
data. We try three alternative weighting schemes: (1) "Geographical" weighting,
by geography, age and presence/absence of a partner; (2) age and highest educational
qualification; and (3) for women only, age, highest qualification, and age at 
first live birth (AFLB). Figure \@ref(fig:plot-res-wt) plots selection effects among the
entire sample, estimated with the three weighting schemes, and also with uniform
weights. Effects for `r res_sig_all` out of 33 polygenic scores are significant
at $p < 0.05/33$ under all four weighting schemes, with `r res_sig_any` scores
reaching significance under at least one scheme. Weighting makes
a large difference to estimates: mean effect sizes across all
polygenic scores are increased by a factor of `r mean_rel$Geographical`
(geographical weighting), `r mean_rel[["Age/Qualification"]]`
(age/qualification) or `r mean_rel[["Age/Qual/AFLB"]]` (age/qualification/AFLB).
Estimates might be further affected by weighting on other demographic variables.

We also check for balancing and diversifying selection by estimating
\@ref(eq:regression) with a quadratic term. In particular, we find diversifying
selection for educational attainment polygenic scores: at higher values of these
scores, the negative effect on fertility is smaller (Appendix Figure
\@ref(fig:plot-purifying)).


```{r plot-res-wt, fig.cap = "Selection effects: weighted and unweighted regressions.  Each point represents a single bivariate regression of number of children on a polygenic score."}

n_regs <- as.double(length(score_names))

res_wt_combined %>% 
      mutate(
        Weights  = fct_relevel(Weights, "Geographical", "Age/Qualification",
                    "Age/Qual/AFLB", "None"),
        term     = pretty_names(term),
        term     = fct_reorder(term, estimate, order_abs(1))
      ) %>% 
      rename(p = p.value) %>% 
      # can't use standard_ggplot because we have non-standard geom_point()
      ggplot(aes(
        x      = estimate, 
        y      = term, 
        yend   = term,
        shape  = Weights, 
        fill   = Weights, 
        colour = p < 0.05/{{n_regs}}
      )) +
      my_vline +
      geom_point(size = 1.5, stroke = 0.5) +
      confint_geom_segment() +
      pval_scale() +
      scale_fill_manual(values = c(
        "None"              = "grey15", 
        "Geographical"      = "dodgerblue4",
        "Age/Qualification" = "steelblue2",
        "Age/Qual/AFLB"      = "darkseagreen4"
      )) +
      scale_shape_manual(values = c(
        "None" = "square filled", 
        "Geographical" = "circle filled",
        "Age/Qualification" = "circle filled", 
        "Age/Qual/AFLB" = "circle filled"
      )) + 
      labs(x = "Effect size", y = "") +
      scale_linetype_manual(name = "", values = c("95% c.i." = 1)) +
      my_theme()


```


To understand the reasons for the ascertainment bias in our sample, and to learn 
about the mechanisms underlying natural selection, we split the sample, starting with
basic demographic variables including education, income and sex. These are all
potential sources of ascertainment bias, as mentioned above. We then re-estimate
\@ref(eq:regression) within each subgroup.

We emphasize that this exercise does
*not* apportion the total correlation with fertility into subgroups. 
That is because the polygenic score may also affect the probability of being in 
each subgroup. For example, the EA3 polygenic score for educational attainment not only affects
fertility among university graduates, and among non-graduates; it also changes
a person's chance of attending university. Nevertheless, differences in equation
\@ref(eq:regression) across subgroups can inform us about how natural
selection works, by constraining the set of possible mechanisms.


```{r plot-income-educ-level, fig.cap = "Selection effects by education and income.", fig.subcap = c("Age left full-time education", "Household income"), fig.ncol = 1}

drake::loadd(res_edu)
n_regs <- as.double(nrow(res_edu))


res_edu %>% 
      mutate(
        "Age left FTE" = fct_relevel(age_fte_cat, "< 16", "16-18", "> 18"),
        "Age left FTE" = add_n(`Age left FTE`, "age_fte_cat")
      ) %>% 
      standard_ggplot(n_regs = n_regs, fill_col = `Age left FTE`, 
                        fill_direction = -1)
   
drake::loadd(res_income)

n_regs <- as.double(nrow(res_income))
res_income %>% 
      mutate(
        Income = factor(income_cat, 
          labels = c("< £18K", "£18-30K", "£31-51K", "£52-100K", "> £100K")
        ),
        Income = add_n(Income, "income_cat")
      ) %>% 
      standard_ggplot(n_regs = n_regs, fill_col = `Income`, 
                        fill_direction = -1)
```


```{r calc-income-edu-controlled}

drake::loadd(res_income_controlled, res_edu_controlled)
n_regs_ic <- nrow(res_income_controlled)
n_sig_ic <- sum(res_income_controlled$p.value < 0.05/n_regs_ic)

n_regs_educ <- nrow(res_edu_controlled)
n_sig_educ <- sum(res_edu_controlled$p.value < 0.05/n_regs_educ)
```


Figure \@ref(fig:plot-income-educ-level) plots selection effects for each
polygenic score, grouping respondents by age of completing full-time education,
and by household income. Effects are larger and more significant for the lowest
income category, and for the lowest education category.  These results could be
explained by age, if older respondents have lower income and are less educated,
and also show more natural selection on polygenic scores. However, when we rerun
the regressions, interacting the polygenic score with income category and also
with a quadratic in age, the interaction with income remains significant at
$p < 0.05/`r n_regs_ic`$ for `r n_sig_ic` out of `r n_regs_ic` regressions. Similarly
if we interact the PGS with age of leaving full time education and a quadratic
in age, the interaction with age leaving full time education remains significant
at $p < 0.05/`r n_regs_educ`$ for `r n_sig_educ` out of `r n_regs_educ` regressions.

Selection effects are also different between men and women (Appendix Figure
\@ref(fig:plot-sexes)). Differences are particularly large for educational
attainment, height and MDD. Several polygenic scores for mental illness and
personality traits are more selected for (or less against) among women,
including major depressive disorder (MDD), schizophrenia and neuroticism, while
extraversion is more selected for among men. Scores for waist circumference and
waist-hip ratio are less selected for among women. One
possible reason for these sex differences is that polygenic scores may affect
fertility via success in marriage markets, and men and women may value different
characteristics in these. 


```{r calc-age-flb-cross}

drake::loadd(res_age_flb_cross)

cor_age_flb <- res_age_flb_cross %>% 
      mutate(category = gsub(":.*", "", term)) %>% 
      pivot_wider(score_name, names_from = category, values_from = estimate) %>%              {cor(.$`age_flb_cat10-22`, .$`age_flb_cat28-52`)}

```


We next focus on variables related to household type and reproductive
strategy. We split males and females by lifetime number of sexual
partners, at the median value of 3 (Figure \@ref(fig:plot-n-partners)a).
For both sexes, selection effects are larger and more significant among
those with more than 3 lifetime partners. Next we split respondents by whether
they were living with a spouse or partner at the time of interview.
Effects are larger among those not living with a spouse or partner
(Figure \@ref(fig:plot-n-partners)b). 

Lastly, we split female respondents by age at first live birth
(AFLB).[^aflb-no-men] There is evidence for genetic effects on AFLB
[@Barban_2016], and there is a close link between this variable and number of
children born. Figure \@ref(fig:plot-age-flb)a shows effect sizes estimated
separately for each tercile of AFLB. Several effects are strikingly different
across terciles. ADHD and MDD are selected for amongst the youngest third of
mothers, but selected against among the oldest two-thirds. Educational
attainment is selected for among the oldest two-thirds of mothers, but is not
significantly selected among the youngest third. Similarly, several polygenic
scores for body measurements are selected against only among older mothers. The
correlation between effect sizes for the youngest and oldest terciles is 
`r cor_age_flb`. 

[^aflb-no-men]: This information is unavailable for men.


```{r plot-n-partners, fig.cap = "Selection effects by number of sexual partners and household type.", fig.subcap = c("Lifetime number of sexual partners", "Household type"), fig.ncol = 1, fig.align = "center", fig.height = 4}

sexes_theme <-  my_theme(
                  plot.margin          = unit(c(1, 1, 0, 1), "lines"),
                  legend.position      = "bottom", 
                  legend.box.spacing   = unit(0, "lines"),
                  legend.box.margin    = margin(-5, 0, 0, -85),
                  legend.margin        = margin(0, 0, 0, 0),
                  legend.box.just      = "left",
                  legend.justification = "left",
                  legend.spacing       = unit(1, "lines")
                )
sexes_guides <- guides(
                  # fill  = guide_legend(ncol = 1),
                  # color = guide_legend(ncol = 1)
                )

drake::loadd(res_partners)
res_partners %<>%       
      filter(grepl(":", term)) %>% 
      mutate(`N partners` = ifelse(grepl("TRUE:", term), "3 or less", "4 or more")) 


n_regs <- as.double(nrow(res_partners))
res_partners %>% 
      mutate(
        score_name = pretty_names(score_name)
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(
        y      = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        yend   = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        x      = estimate, 
        fill   = `N partners`, 
        colour = p < 0.05/{{n_regs}}
      )) + 
        my_vline +
        confint_geom_segment() +
        pval_geom_point() +
        facet_wrap(vars(sex), scales = "free_y") +
        my_fill_scale(n = 2) +
        pval_scale() +
        sexes_theme +
        my_labs(x = "", y = "") + 
        scale_y_reordered() +
        scale_linetype_manual(name = "", values = c("95% c.i." = 1)) +
        sexes_guides

drake::loadd(res_with_partner_sex)
res_with_partner_sex %<>% filter(grepl(":", term)) 

n_regs <- as.double(nrow(res_with_partner_sex))

res_with_partner_sex %>% 
      mutate(
        score_name = pretty_names(score_name),
        Household = ifelse(grepl("TRUE", term), "With partner", 
                      "Without partner")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(
        y      = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        yend   = reorder_within(score_name, estimate, sex, fun = order_abs(1)), 
        x      = estimate, 
        fill   = Household, 
        colour = p < 0.05/{{n_regs}}
      )) + 
        my_vline +
        confint_geom_segment() +
        pval_geom_point() +
        facet_wrap(vars(sex), scales = "free_y") +
        my_fill_scale(n = 2) +
        pval_scale() +
        sexes_theme +
        my_labs(x = "", y = "") + 
        scale_y_reordered() +
        scale_linetype_manual(name = "", values = c("95% c.i." = 1)) +
        sexes_guides

```


```{r calc-corr-age-flb}

drake::loadd(res_age_flb)
drake::loadd(res_sex)

res_sex_comparison <- res_sex %>% 
      filter(sex == "Female") %>% 
      mutate(score_name = term) %>% 
      select(score_name, term, estimate:conf.high)

res_age_flb %<>% filter(term != "age_flb") 

res_combined <- bind_rows(
        raw      = res_sex_comparison, 
        with_age = res_age_flb %>% select(score_name, term, estimate:conf.high), 
        .id      = "reg.type"
      ) %>% 
      arrange(score_name)

raw_flb_comparison <- res_combined %>% 
      pivot_wider(
        id_cols     = score_name, 
        names_from  = reg.type, 
        values_from = estimate
      )

cor_raw_flb <- cor(raw_flb_comparison$raw, raw_flb_comparison$with_age)
opp_signed <- sum(
        sign(raw_flb_comparison$raw) != sign(raw_flb_comparison$with_age)
      )

```


To investigate this further, we estimate equation \@ref(eq:regression)
*controlling* for AFLB, again among females (Figure \@ref(fig:plot-age-flb)b).
In `r opp_signed` out of `r length(score_names)` cases, effects change sign when
controls are added. The correlation between effect sizes controlling for AFLB,
and raw effect sizes, is `r cor_raw_flb`. Thus, selection effects seem to come
through two opposing channels: an effect on AFLB, and an opposite-signed effect
on number of children controlling for AFLB. Again, we do not claim that AFLB
is exogenous to an individual's polygenic scores. Indeed, our analyses show
that it is not (Appendix Figure \@ref(fig:plot-age-flb-dv)). Rather, we argue
that AFLB mediates part of the relationship between polygenic scores and
fertility, and that once this is controlled for, the remaining part of the
relationship has the opposite sign.


```{r plot-age-flb, fig.cap = "Selection effects by age at first live birth, and controlling for age at first live birth (women only).", fig.ncol = 1,  fig.subcap = c("Effect sizes by age at first live birth terciles (women only)", "Effect sizes controlling for age at first live birth. Effect sizes without controls (for women only) shown for comparison."), fig.align = "center"}

drake::loadd(res_age_flb_cross)

# this counts each cross term as a separate test:
n_regs <- as.double(nrow(res_age_flb_cross)) 

res_age_flb_cross %>% 
      mutate(
        `Age at first live birth` = gsub("age_flb_cat(.*):.*", "\\1", term),
        `Age at first live birth` = add_n(`Age at first live birth`, "age_flb_cat")
      ) %>% 
      standard_ggplot(score_col = score_name, 
                      fill_col = `Age at first live birth`, n_regs = n_regs,
                      fill_direction = -1)

n_regs <- as.double(nrow(res_age_flb))

res_combined %>% 
      mutate(
        score_name = pretty_names(score_name),
        score_name = fct_reorder(score_name, estimate, order_abs(2)),
        Regression = ifelse(reg.type == "raw", "Uncontrolled", "Controlled for AFLB")
      ) %>% 
      standard_ggplot(score_col = score_name, 
                      fill_col = Regression, n_regs = n_regs)


```


\FloatBarrier

## Selection in the parents' generation

The UK Biobank data contains information on respondents' number of
siblings. Since respondents' polygenic scores are equal in expectation
to the mean scores of their parents, we can use this to look at selection 
effects in the parents' generation. We estimate equation (\ref{eq:regression})
using *number of siblings* (including the respondent) as the dependent variable.
The parents' generation has an additional source of ascertainment bias: sampling
parents of respondents overweights parents who have many children. For instance,
parents of three children will have, on average, three times more children
represented in UK Biobank than parents of one child. Parents of no children will
by definition not be represented. To compensate, we reweight our preferred
weightings (Age/Qualification) by the inverse of *number of siblings*. 

Selection effects are highly correlated across the two generations, and most
share the same sign (Appendix Figure \@ref(fig:plot-res-sibs)). Absolute effect
size estimates are larger for the parents' generation. We treat this result
cautiously, because when we split respondents up by year of birth, we find few
differences in effect sizes between early- and late-born respondents, for either
generation. In other words, since estimated effect sizes change across
"generations", but do not change over time within either generation, the change
may be due to remaining ascertainment bias within the sample or other effects.
In particular, effect sizes in both generations may depend on polygenic scores'
correlation with childlessness, and we cannot estimate this for the parents'
generation.

Although the direction of selection effects does not change between the
generations, there are other differences. Compared to our standard polygenic
scores (residualized on 100 principal components of genetic data), 
selection effects on unresidualized scores are about ten percent higher
on average for *number of siblings*, whereas effects for *number of children*
barely change (Appendix Figure \@ref(fig:plot-regs-controlled-pcs)). This could
be because earlier fertility is driven more by geographically clustered
deprivation [e.g. via an insurance motive, @rendall1998old], which may correlate
with the broad-scale genetic variation captured by principal components, such as
ancestry.

We also check whether selection effects differ by socio-economic status in the
parents' generation. We have no information about parents' income, so we use the
1971 Townsend deprivation score of respondents' birthplace as a proxy
[@townsend1987deprivation]. Results (Appendix Figure
\@ref(fig:plot-siblings-townsend)) show the same pattern as for respondents:
effect sizes are larger and more often significant in the most deprived areas.


```{r calc-siblings-children}
sib_chn_cor <- cor.test(famhist$n_sibs, famhist$n_children, use = "complete")
stopifnot(sib_chn_cor$p.value < 2e-16)
```


Lastly, the siblings data lets us check for a "quantity-quality tradeoff" between
number of children and number of grandchildren (for the parents' generation). We
do not find any: in fact, the correlation between *number of siblings* and 
*number of children* is positive ($\rho$ = `r sib_chn_cor$estimate`, 
$p < 2 \times 10^{-16}$).


## Human capital and natural selection

These results show that selection effects are weaker, absent, or even reversed
among some subgroups of the population. A possible explanation for this comes 
from the economic theory of fertility 
[@becker1960economic; @willis1973new;@becker1976child]. According to this theory,
increases in a person's wage affect their fertility via two opposing channels.
There is an *income effect* by which children become more affordable, like any
other good. There is also a *substitution effect*: since childrearing has a cost
in time, the opportunity cost of childrearing increases if one's market wage is
higher. The income effect leads higher earners to have more children. The
substitution effect leads them to have fewer children. 

Suppose that certain genetic variants correlate with human capital -- skills or
other characteristics that affect an individual's earnings in the labour market
[@mincer1958investment; @becker1964human]. These variants may then be associated
with opposing effects on fertility. The income effect will lead to natural
selection in favour of earnings-increasing variants (or variants that are merely
associated with higher earnings). The substitution effect will do the reverse.

To show this, we write down a simple model of fertility choices. $h$ is
an individual's level of human capital. For now, we simply identify this with
his or her wage $W$. Raising a child takes time $b$. People maximize utility
from the number of children $N$, and income $Y\equiv(1-bN)W$. An individual's
payoff is

\[
U = u(Y)+aN.
\]

Here $a$ captures the strength of preference for children. $u(\cdot)$ captures
the taste for income, and is concave and increasing. We treat $N$ as continuous, in line with
the literature: this can be thought of as the expected number of children
among similar people. The marginal benefit of an extra child is
$\frac{dU}{dN} = -bWu'(Y)+a$. The effect of an increase in human capital on this
marginal benefit is

\[
\frac{d^{2}U}{dNdh}=\underbrace{-bu'(Y)}_{\textrm{Substitution effect}}\underbrace{-bYu''(Y)}_{\textrm{Income effect}}.
\]

The *substitution effect* is negative and reflects that when wages increase,
time devoted to childcare costs more in foregone income. The positive *income
effect* depends on the curvature of the utility function, and reflects that when
income is higher, the marginal loss of income from children is less painful.

To examine education and fertility timing, we extend the model to
two periods. For convenience we ignore time discounting, and assume
that credit markets are imperfect so that agents cannot borrow. Write

\begin{equation}
U(N_{1},N_{2}) = u(Y_{1}) + u(Y_{2}) + aN_{1} + aN_{2}\label{eq:U}
\end{equation}

Instead of identifying human capital with wages, we now allow
individuals to choose a level of education $s \in [0,1]$, which has
a time cost in period 1. Education is
complementary to human capital $h > 0$, and increases period 2 wages, which take 
the  simple functional form $w(s,h) = sh$. We normalize period 1 wages
to 1. Letting $u(\cdot)$ take the constant relative risk aversion (CRRA) form
$u(y)=\frac{y^{1 - \sigma} - 1}{1 - \sigma}$, where $\sigma>0$ captures risk aversion
in income, we examine total fertility $N^{*} = N_{1}^{*} + N_{2}^{*}$ and the 
*fertility-human capital relationship*, $\frac{dN^{*}}{dh}$. For $\sigma < 1$
in a neighbourhood of $\sigma = 1$, Table \@ref(tab:theory) shows five theoretical 
predictions, along with corresponding empirical results for the correlation between
polygenic scores and fertility.[^one-period-results]

[^one-period-results]: Predictions 1-3 also hold in the one-period model. Our empirical 
results are actually stronger than prediction 5, in that
correlations with fertility are *reversed* at higher AFLB. This prediction can be
accommodated in the model if children have a money cost as well as a time cost. 
Appendix Figure \@ref(fig:N-plot-with-m) shows an example.


------------------------------------------------------------------------------------
    **Theory**                                        **Empirical results**
--- ------------------------------------------------- ------------------------------
1.  The fertility-human capital                       Figures \@ref(fig:plot-means-over-time) 
    relationship is negative:                         and \@ref(fig:plot-res-wt).
    $\frac{dN^{*}}{dh} < 0$.        
            
2.  The fertility-human capital                       Figure \@ref(fig:plot-income-educ-level)a. 
    relationship is weaker (closer to zero)           Selection effects are also weaker at higher
    at higher wages and/or levels of                  polygenic scores for 
    human capital.                                    educational attainment (Appendix Figure
                                                      \@ref(fig:plot-purifying)).
            
3.  The relationship is more negative when            Stronger effects for single parents
    the time burden of children, $b$, is              (Figure \@ref(fig:plot-n-partners)).
    larger.    
            
4.  The relationship is weaker at higher              Figure \@ref(fig:plot-income-educ-level)b.
    levels of education $s$.    
            
5.  The relationship is weaker among those            Effects weaker among those starting 
    who start fertility in period 2                   fertility later 
    ($N_{1}^{*} = 0$) than among those who            (Figure \@ref(fig:plot-age-flb)a).
    start fertility in period 1 
    ($N_{1}^{*} > 0$).
----------------------------------------------------------------------------------------

Table: (\#tab:theory) Predictions from the theoretical model and corresponding 
empirical results.


```{r literature, eval = FALSE}
"Other models of the literature

## LITERATURE NOTES

* Lundberg and Pollak JEP 2007:
    - several models where nonmarital fertility is more likely for poorer women
    - Burdett and Ermisch 2002, Willis 1999, Neal 2004
    - Unmarried parents may not be able to credibly commit to each other
  
* Happel et al. 1984
   - childbirth is chosen to minimize loss of earnings from interrupted work.
   - early childbirth if your skills decay a lot  while childrearing. (No point
     gaining them then losing them.) Late childbirth otherwise.
   - late childbirth if you have acquired skills before marriage.
   - Capital-market imperfections make it wise to delay childbirth (till man has
     high income to cope while wife is at home). More so for men's wages which
     rise faster.
   - empirically: (among married couples) 
     - high-skill occupation women have children later. Higher-earning men have 
     them earlier.
     
- Caucutt et al. 2001
  - high waged and high educated women (of recent cohorts) more likely to
    delay childbirth
    
- Cohen et al. 2013
  - analyse effect of Israeli subsidy
  - along the way, find that income affects fertility negatively at low income,
    positively at high income, supporting Becker and Tomes 76

- Becker and Tomes 1976
  - investment in children quality and quantity when quality must be equal
    across children; and some quality comes from an innate endowment.
  - "observed" quantity-of-children income elasticity rises as income rises
  - while "observed" quality-of-children elasticity falls as income rises, 
    b/c more of total quality must be paid for by parents themselves. 
  - the need to pay more for quality results in lower quantity, since total cost is 
    quantity X quality
  - something fishy about argument somehow...
  
- Lindo 2008
  - men who lose a job have children sooner, but fewer children in total
  - controls for all skills effects
  - so overall effect of income is positive, but timing changes
  
- Monstad et al. 2008
  - using an education reform to vary education
  - no evidence of a causal effect on fertility, but births start later
  * maybe the usual correlation reflects underlying skills. Your data could
    test this.
    
- Autor et al. 2017
  - the China shock (exogenous shock to income) 
  - they use Chinese import penetration by industry (instrumented using
    other countries) times industry employment shares; and separate it for
    males and females
  - shocks to male employment decreased fertility and marriage; 
    increased unmarried mothers
  - shocks to female employment *increase* marriage and motherhood
  * this roughly seems to be the consensus
  * Whereas, you find that male and female *skills* (as 'instrumented' by PGS)
    have the same effect on fertility - i.e. less skills = higher fertility
 
- Baudin et al. 2015
  * look at childlessness vs. fertility (extensive/intensive margin)
  * some people are too poor to afford children
  * 'a husband alleviates the time cost of having children'. But for some groups (see appendix)
    there is a +ve income/fertility gradient
  * fertility decreases with female educn due to subst effect
  * baby boom and bust in US: a decline in childlessness caused by poverty, then an increase
    caused by substn effect ('opportunity cost')
  * fertility decreases with educn both for single mothers and married women
    - slightly (25%) higher decrease per year of educn (Table 2) for single mothers




```


The above does not aim to provide a complete theory of fertility. Rather, it
shows that a relatively simple economic model can explain many of our results.
Other empirical work in economics also supports our mechanism.
@caucutt2002women and @monstad2008education show that education and skills
affect age at first birth and fertility. Income decreases fertility at low
income levels, but increases it at higher income levels [@cohen2013financial].
US fertility decreases faster with education among single mothers than married
mothers [@baudin2015fertility], in line with our prediction 3 and as predicted by
@becker1981treatise.


```{r plot-cors-earnings-educ, fig.cap="Selection effects by correlations with earnings and educational attainment. Each point represents one polygenic score."}

drake::loadd(res_all)
drake::loadd(res_age_flb)
drake::loadd(ashe_income)

famhist <- add_ashe_income(famhist, ashe_income)

cors_income <- cor(famhist[score_names], famhist$first_job_pay, use = "pairwise")
# TODO: use age_left_edu when this includes college

cors_educ <-   cor(famhist[score_names], famhist$age_fulltime_edu, use = "pairwise")

effect_size <-  res_all %>% 
                  filter(dep.var == "N children", reg.type == "controlled") %>% 
                  pull(estimate)

effect_size_flb <-  res_age_flb %>% 
                      filter(term != "age_flb") %>% 
                      pull(estimate)

dfr <- data.frame(
        PGS          = rep(score_names, 4),
        Correlation  = rep(c("Earnings", "Education"), each = 33, 2),
        cor          = rep(c(cors_income[,1], cors_educ[,1]), 2),
        effect_size  = c(rep(effect_size, 2), rep(effect_size_flb, 2)),
        Controls     = rep(c("None", 
                        "Age at first live birth"), each = 66)
      )

dfr %>% 
      mutate(
        Controls = fct_relevel(Controls, "None"),
        PGS      = ifelse(abs(effect_size) > 0.035, PGS, ""),
        PGS      = sub("(^\\w*?)_.*", "\\1", PGS)
      ) %>% 
      filter(
        Controls == "None"
      ) %>% 
      ggplot(aes(effect_size, cor, colour = Correlation)) + 
        geom_point() +
        geom_text(aes(label = PGS), size = 3, colour = "black", 
                    check_overlap = TRUE, nudge_x = 0.005) +
        facet_wrap( ~ Correlation, scales = "free") + 
        my_vline + 
        my_hline +
        labs(x = "Selection effect", y = "Correlation") +
        coord_cartesian(clip = "off") +
        scale_colour_brewer(type = "qual", palette = 2) +
        theme(
          legend.position = "none",
          panel.spacing = unit(20, "pt")
        )

```


## Testing the theory

We test this explanation in two ways. First, the economic theory predicts that
genetic variants will be selected for (or against) in proportion to their
correlation with human capital. Figure \@ref(fig:plot-cors-earnings-educ) plots
selection effects on each polygenic score against that score's correlation with
two measures of human capital: earnings in a respondent's first job, and
educational attainment. The relationships are strongly negative. Thus, human
capital appears to be relevant to natural selection. Substitution effects
dominate income effects overall, which fits the known association between income
and lower fertility [@becker1960economic;@jones2006economic]. The correlations
reverse when we control for age at first live birth, suggesting that within AFLB
categories, the income effect dominates. (See also the Supplementary Animations
for the uncorrected effects per AFLB group, which show the correlation reversing
after age 22.)

Second, we re-estimate equation \@ref(eq:regression) for each polygenic score,
controlling for education levels. Effect sizes are
generally reduced (Appendix Figure \@ref(fig:plot-res-educ)), with
large reductions for polygenic scores for educational attainment. This 
suggests that education indeed mediates the effect of polygenic scores on fertility.

```{r calc-fe-fertility}
drake::loadd(res_fe_fertility)
drake::loadd(res_all)

res_all %<>% filter(dep.var == "N children", reg.type == "controlled")
res_fe_fertility %<>% filter(Regression == "Raw")
res_all %<>% left_join(res_fe_fertility, by = "term")

lm_pool_w <- tidy(lm(estimate.y ~ estimate.x, res_all))
coef_pool_w <- lm_pool_w[[2, "estimate"]]
se_pool_w <- lm_pool_w[[2, "std.error"]]
```

These results support our theory that natural selection on polygenic scores is
driven by their correlation with human capital. This need not imply
that a given polygenic score is causal of either fertility or human capital. Instead
scores may simply correlate with causally relevant environments, for example via
"genetic nurture" effects [@kong2018nature] or population stratification. To 
investigate the presence of these indirect genetic effects, we estimate equation
\@ref(eq:regression) among full siblings, controlling for a full set of sibling
group dummies (Appendix Figure \@ref(fig:plot-res-fe-fertility)). With a reduced
sample size, all within-sibling effects are insignificant after Bonferroni
correction. However, effect sizes are positively correlated with effect sizes
from the pooled model, and about 1/3 smaller (regressing within-sibling on
pooled effect sizes, $b$ = `r coef_pool_w`). This attenuation is broadly in line
with the ~50% decrease in heritability in within-sibling GWASs on age at first
birth and educational attainment [@Howe2021.03.05.433935]. We see these results
as providing weak evidence that polygenic scores cause fertility, with effects
being at least partly driven by gene-environment correlations. We also reran
within-siblings regressions adding a control for education. Most effect sizes
barely change, suggesting that our measure of education does not in general
mediate differences in fertility among siblings. However, for educational
attainment polygenic scores, effect sizes are reduced by about 20%.


```{r calc-risk-attitude}

drake::loadd(res_risk_control)
drake::loadd(res_all)

n_risk_sig <- res_risk_control %>% 
            filter(term == 'f.2040.0.0') %>% 
            {sum(.$p.value <= .Machine$double.eps)}

n_pgs_sig <- res_risk_control %>% 
            filter(term == score_name) %>% 
            {sum(.$p.value <= 0.05/33)}

res_compare <- res_all %>% 
  filter(dep.var == "N children", reg.type == "controlled") %>% 
  left_join(res_risk_control %>% filter(term == score_name), by = "score_name")

ratio <- res_compare$estimate.y/res_compare$estimate.x
n_less_sig <- sum(res_compare$p.value.y <= 0.05/33 & res_compare$p.value.x > 0.05/33)
stopifnot(n_less_sig == 0)
```

An alternative theory is that polygenic scores correlate with the motivation to
have children (parameter $a$ in the model; cf. @jones2008fertility). However, this
theory would not explain why our results are weaker at higher incomes and
education levels.[^note] Another alternative is that traits under selection are
linked to externalizing behaviour and risk-seeking. This might be partially
captured by our risk-aversion parameter $\sigma$, which affects fertility by
changing the marginal utility of income, but a more direct channel is risky
sexual behaviour [@mills2021identification]. The data here provide some support
for this: scores which might plausibly be linked to externalizing behaviour,
like ADHD and younger age at smoking initiation, are selected for. However,
this theory is less good at explaining variation in selection across the full
range of scores, including physical measures such as waist-hip ratio and BMI. We
test this theory directly by re-estimating equation \@ref(eq:regression)
controlling for a measure of risk attitude (UK Biobank field 2040).  The median
ratio of effect sizes between regressions with and without controls is 
`r median(ratio)`; all scores which are significant at $p < 0.05/33$ in
uncontrolled regressions remain so when controlling for risk attitude. This
non-result could simply reflect the imprecision of the risk aversion measure,
which is a single yes/no question. However, the risk aversion measure *does*
predict the overall number of children, highly significantly ($p < 2 \times
10^{-16}$ in `r n_risk_sig` out of 33 regressions). Given that, and the statistical power
we get from our sample size, we believe that the non-result is real: while risk
aversion does predict fertility in the sample, it is not an important channel
for natural selection.

[^note]: Indeed in the one-period model, while $\frac{dN^{*}}{da} > 0$,
this effect becomes stronger, not weaker, at higher wages.


# Discussion

Previous work has documented natural selection in modern populations on variants
underlying polygenic traits [@beauchamp2016genetic; @kong2017selection;
@sanjak2018evidence]. We show that correlations between polygenic scores and
fertility are highly concentrated among specific subgroups of the population,
including people with lower income, lower education, younger first parenthood,
and more lifetime sexual partners. Among older mothers (22+), selection effects
are reversed. Furthermore, the size of selection effects on a polygenic score
correlates with that score's association with labour market earnings. The
economic theory of fertility provides a parsimonious explanation for these findings.
Because of the substitution effect of earnings on fertility, scores are selected
for when they correlate with low human capital, and this effect is stronger at 
lower levels of income and education.

Polygenic scores which correlate with high (low) earnings and more (less)
education are being selected against (selected for). In addition, many of the
phenotypes under positive selection are linked to disease risk. Many people
would probably prefer to have high educational attainment, a low risk of ADHD
and major depressive disorder, and a low risk of coronary artery disease, but
natural selection is pushing against genes associated with these traits.
Potentially, this could increase the health burden on modern populations, but
that depends on effect sizes. Our results suggest that naïve estimates can be
affected by sample ascertainment bias. This problem may be less serious in
surveys which aim to be representative (as the UK Biobank does not). However,
there is still scope for bias, since not all respondents consent to the
collection of genetic data. For instance, completion rates for genotype data in
the US Health and Retirement Study were around 80-85% [@hrs2020]. Researchers
should be aware of the risks of ascertainment when studying modern natural
selection.

We also do not know how estimated effect sizes of natural selection will change
as more accurate measures of genetic variation are produced. And we are unsure
whether genetic variants underlying other phenotypes will show a similar pattern
of natural selection to those studied here. In addition, genetic effects on
educational attainment have been shown to be inflated in population-based
samples as compared to within-family designs, likely because of indirect genetic 
effects, gene-environment correlations, and/or assortative mating 
[@lee2018gene; @selzam2019comparing; @kong2018nature; @Howe2021.03.05.433935]. 
In short, it is probably too early to tell whether modern natural selection has a
substantively important effect on the genetic make-up of the population.
Nevertheless, we note that selection effects on our measured polygenic scores
are still relatively small, even after reweighting to account for ascertainment
bias.

```{r calc-inequality}
drake::loadd(res_ineq)

ineq_comp <-  left_join(
                res_ineq %>% filter(income_cat == 1),
                res_ineq %>% filter(income_cat == 5),
                by = "score"
              )

ineq_comp %<>% mutate(
                 diff_children = wtd.y - wtd.x,
                 diff_hypoth   = unw.y - unw.x,
                 pct_change    = 100 * (diff_children - diff_hypoth)/diff_hypoth
               )

median_pct_change <- median(ineq_comp$pct_change)
n_increases <- sum(ineq_comp$pct_change > 0)

```

Because selection effects are concentrated in lower-income groups, they may also
increase inequality with respect to polygenic scores. For example, Figure
\@ref(fig:plot-mean-EA3-by-income) graphs mean polygenic scores for educational
attainment (EA3) among children from households of different income groups. The
blue bars show the actual means, i.e. parents' mean polygenic score weighted by
number of children. The grey bars show the hypothetical means if all households
had equal numbers of children. Natural selection against genes associated with
educational attainment, which lowers the mean, is stronger at the bottom of the
income distribution, and this increases the differences between groups. Overall,
natural selection increases inequality for `r n_increases` out of 33 polygenic
scores, with a median increase of `r median_pct_change`% in the difference
between highest and lowest income groups (Appendix Table
\@ref(tab:tbl-inequality-effects)). If inequalities in polygenic scores are
important for understanding class mobility and structure [@belsky2018genetic;
@Rimfeld_2018], then these increases are substantially important. Since many
polygenic scores are predictive of disease risk, they could also potentially
increase health inequalities. In general, the evolutionary history of
anatomically modern humans is related to disease risk [@benton2021influence];
understanding the role of contemporary natural selection may aid in mapping the
genetic architecture of current health disparities.


```{r plot-mean-EA3-by-income, fig.align = "center", fig.cap = "Mean polygenic score for educational attainment (EA3) of children by household income group. Blue is actual. Grey is hypothetical in the absence of selection effects. Respondents weighted by Age/Qualification.", fig.width = 4}
income_EA3 <- famhist %>% 
      filter(! is.na(n_children), ! is.na(income_cat)) %>% 
      mutate(
        `Household income` = factor(income_cat, 
                          labels = c("< £18K", "£18-30K", "£31-51K", "£52-100K", 
                                      "> £100K"))
      ) %>% 
      group_by(`Household income`) %>% 
      summarize(
        `Without selection` = mean(EA3_excl_23andMe_UK, na.rm = TRUE),
        Actual              = weighted.mean(EA3_excl_23andMe_UK, n_children, 
                                              na.rm = TRUE)
      ) %>% 
      tidyr::pivot_longer(-`Household income`) %>% 
      rename(
        `Children's mean EA3` = value
      )

income_EA3 %>% 
              ggplot(aes(`Household income`, `Children's mean EA3`, fill = name)) +
                geom_col(position = "dodge") +
                scale_fill_manual(values = c("navy", "grey60")) +
                theme(
                  legend.background = element_rect(fill = "white", colour = NA),
                  legend.title = element_blank(), 
                  legend.position = c(.25, .85)
                )
```


Existing evidence on human natural selection has led some to "biocosmic
pessimism" [@sarraf2019modernity]. Others are more sanguine, and argue that
natural selection's effects are outweighed by environmental improvements, such
as those underlying the Flynn effect [@flynn1987massive]. The evidence
here may add some nuance to this debate. Patterns of natural selection have been
relatively consistent across the past two generations, but they are not the
outcome of a universal, society-wide phenomenon. Instead they result from 
opposing forces, operating in different parts of society and pulling in
different directions.

Any model of fertility is implicitly a model of natural selection, but so far,
the economic and human genetics literatures have developed in parallel.
Integrating the two could deepen our understanding of natural selection in
modern societies. Economics possesses a range of theoretical models on the
effects of skills, education and income [see @hotz1997economics;
@lundberg2007american]. One perennial problem is how to test these theories in a
world where education, labour and marriage markets all interact. Genetic data,
such as polygenic scores, could help to pin down the direction of causality,
for example via Mendelian randomization [@davey2003mendelian].
Conversely, theory and empirical results from economics can shine a light on the
mechanisms behind natural selection, and thereby on the nature of individual
differences in complex traits and disease risk.


# Materials and methods

We use participant data from UK Biobank [@bycroft2018uk], which has received
ethical approval from the National Health Service North West Centre for Research
Ethics Committee (reference: 11/NW/0382). We limit the sample to white British
participants of European descent, as defined by genetic estimated ancestry and
self-identified ethnic group [@abdellaoui2019genetic], giving a sample size of
`r comma_num(nrow(famhist))`. For regressions on number of children we use
participants over 45, since most fertility is completed by this age. This gives
a sample size of `r comma_num(nrow(famhist %>% filter(kids_ss)))`.

Polygenic scores were computed by summing the alleles across ~1.3 million 
genetic variants weighted by their effect sizes as estimated in 33 genome-wide 
association studies (GWASs) that excluded UK Biobank. To control for population 
stratification, we corrected the polygenic scores for 100 principal components (PCs). 
To compute polygenic scores and PCs, the same procedures were followed as 
described in @abdellaoui2019genetic.

Earnings in first job are estimated from mean earnings in the 2007 Annual Survey
of Hours and Earnings, using the SOC 2000 job code (Biobank field 22617).

Population data for weighting is taken from the 2011 UK Census and the 2006
General Household Survey (GHS). Weighting for Age/Qualification and 
Age/Qualification/AFLB weights was done using marginal totals from a linear
model, using the `calibrate()` function in
the R "survey" package [@lumley2020survey]. Geographical weighting was done with
iterative post-stratification using the `rake()` function, on Census Middle Layer
Super Output Areas, sex and presence/absence of a partner.


# Acknowledgements

AA is supported by the Foundation Volksbond Rotterdam and by ZonMw grant 
849200011 from The Netherlands Organisation for Health Research and Development. 
This study was conducted using UK Biobank resources under application numbers 
40310 and 19127.

\FloatBarrier
\clearpage
# Appendix

## Natural selection by sex

```{r plot-sexes, fig.cap = "Selection effects by sex. Highlighted lines are significant differences at p < 0.05/33. Highlighted points are significantly different from 0 at p < 0.05/66."}

drake::loadd(res_sex)
n_regs <- as.double(nrow(res_sex))


res_sex_wide <- res_sex %>% 
      select(term, estimate, sex, diff.p.value) %>% 
      pivot_wider(names_from = "sex", values_from = "estimate") %>% 
      mutate(
        min_est = pmin(Female, Male),
        max_est = pmax(Female, Male),
        term = pretty_names(term),
        term = fct_reorder(term, Female)
      ) 
n_comps <-as.double(nrow(res_sex_wide)) 

res_sex %>% 
      mutate(
        term = pretty_names(term),
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, fill = sex, 
            colour = p < 0.05/{{n_regs}})) +
        my_vline +
        geom_linerange(data = res_sex_wide, 
          aes(
            y      = term,
            x      = NULL,
            fill   = NULL,
            xmin   = min_est, 
            xmax   = max_est, 
            colour = diff.p.value < 0.05/{{n_comps}}
          ), 
        group = "") +
        pval_geom_point() +
        pval_scale(name = "p < 0.05 (corrected)") +
        my_fill_scale(n = 2) +
        my_labs() +
        my_theme()

```


\FloatBarrier
\clearpage
## Weighted regressions

Table \@ref(tab:tbl-res-wt) gives effect sizes as a proportion of the unweighted
effect size, for all polygenic scores which are consistently signed and which
are significantly different from zero in unweighted regressions.


```{r tbl-res-wt}

rel_sizes %>% 
      arrange(desc(Geographical)) %>% 
      bind_rows(rel_sizes_summary) %>% 
      rename(PGS = term) %>% 
      as_hux() %>% 
      insert_row("", "Weighting", fill = "") %>% 
      set_caption(
        "Weighted effect sizes as a proportion of unweighted effect sizes."
      ) %>% 
      merge_cells(1, 2:4) %>% 
      set_align(1, 2, "centre") %>% 
      set_header_rows(1, TRUE) %>% 
      theme_compact() %>% 
      set_number_format(-1, -1, 2) %>% 
      style_headers(bold = TRUE) %>% 
      set_bottom_border(1, 2:4) %>% 
      set_bottom_border(2, everywhere) %>% 
      set_italic(final(2), 1) %>% 
      set_font_size(8) %>% 
      set_width(0.7) %>% # ensures line wrapping below
      set_col_width(c(.34, .22, .22, .22)) %>% 
      add_footnote(paste0(
        "Only consistently-signed and significant (when unweighted) ",
        "estimates are shown. ",
        "Age/Qual/AFLB as a proportion of unweighted regressions ",
        "including females only."),
        top_padding  = 0,
        left_padding = 0
      )
      
```


\FloatBarrier
\clearpage
## Balancing and diversifying selection


We rerun equation \@ref(eq:regression), adding a quadratic term in $PGS_i$,
and using our age/qualification weights. Scores for hip circumference and drinks
per week show significant balancing selection ($p < 0.05/33$, negative
coefficient). Scores for educational attainment (EA2 and EA3) show significant
diversifying selection ($p < 0.05/33$, positive coefficient), which reduces the
strength of selection against educational attainment at very high levels of the 
PGS. Figure
\@ref(fig:plot-purifying) plots predicted number of children against polygenic
score from these regressions. 

We also checked for balancing selection in the parents' generation, using
age/qualification weights multiplied by the inverse of number of siblings. 
Scores for EA2 and EA3 again show significant diversifying selection 
($p < 0.05/33$, positive coefficient). Other scores including hip circumference 
and drinks per week are not significant.


```{r plot-purifying, fig.align = "center", fig.cap = "Purifying/diversifying selection: predicted number of children by polygenic score.", fig.height = 5}

drake::loadd(res_quadratic)
sig <- res_quadratic %>% 
         filter(grepl("^2", term, fixed = TRUE)) %>% 
         filter(p.value < 0.05/33) %>% 
         pull(score_name)

stopifnot(sig == c("dpw_substance_use", "EA2_noUKB", "EA3_excl_23andMe_UK",
                     "hip_combined"))

sig_all_terms <- res_quadratic %>% filter(score_name %in% sig)


drake::loadd(res_sibs_quadratic)
sig_sibs <- res_sibs_quadratic %>% 
         filter(grepl("^2", term, fixed = TRUE)) %>% 
         filter(p.value < 0.05/33) %>% 
         pull(score_name)
stopifnot(sig_sibs == c("EA2_noUKB", "EA3_excl_23andMe_UK"))

op <- par(mfrow = c(2,2))
for (score in sig) {
  coef <- sig_all_terms %>% filter(score_name == score) %>% pull(estimate)
  curve(
         coef[1] + coef[2] * x + coef[3] * x^2,
         xlim = c(-4, 4), 
         xlab = pretty_names(score),
         ylab = "Predicted N children"
       )
}
par(op)

```




\FloatBarrier
\clearpage
## Parents' generation


```{r calc-res-sibs}
drake::loadd(res_wt_age_qual_weights)
drake::loadd(res_children_comparison)

cmp <- left_join(res_wt_age_qual_weights, res_children_comparison, by = "term")

cor_inc_vs_excl_childless <- cor(cmp$estimate.x, cmp$estimate.y)
n_smaller <- sum(abs(cmp$estimate.y) < abs(cmp$estimate.x))

pct_diff <- abs(cmp$estimate.y)/abs(cmp$estimate.x) - 1
median_pct_diff <- median(pct_diff)

```


Figure \@ref(fig:plot-res-sibs) shows regressions of *number of siblings*, i.e.
parents' number of children, on polygenic scores. By definition, members of the
parents' generation who had no children cannot be included in this data. For a
clean comparison with the respondents' generation, we rerun regressions on
*number of children* excluding those with no children, and show
results in the figure.

Excluding childless people in the parents' generation could bias our estimates.
To learn about this, we compare effect sizes excluding and including childless
people in the *current* generation. The correlation
between the two sets of effect sizes is `r cor_inc_vs_excl_childless`. So, patterns across
different scores are broadly similar whether the childless are counted or not. 
However, absolute effect sizes are smaller when the
childless are excluded, for `r n_smaller` out of 33 scores; the median
percentage change is `r percent(median_pct_diff)`. The fact that childless
people have such a strong effect on estimates makes it hard to compare total
effect sizes across generations. In particular, since the parents' generation has
a different distribution of numbers of children, childless people may have had 
more or less effect in that generation.


```{r plot-res-sibs, fig.cap = "Selection effects, respondents' parents vs. respondents."}
drake::loadd(res_sibs_parent_weights)
drake::loadd(res_children_comparison)

res_sibs_kids <-  bind_rows(
                    Siblings = res_sibs_parent_weights, 
                    Children = res_children_comparison,
                    .id = "Dep. var."
                  )

standard_ggplot(res_sibs_kids, `Dep. var.`, n_regs = length(score_names))
```



```{r calc-period-regs}

drake::loadd(res_period_parents)
drake::loadd(res_period_children)

res_period <- bind_rows(res_period_parents, res_period_children)

res_period %<>% 
      tidyr::extract(term, c("period", "score_name"),
        regex="year_split(.*):(.*)")
n_tests <- nrow(res_period)/2


summary_period <- res_period %>% 
      select(-std.error, -statistic) %>% 
      pivot_wider(
        names_from  = period, 
        values_from = c(estimate, conf.low, conf.high, p.value)
      ) %>% 
      group_by(score_name, children) %>% 
      mutate(
        significant = diff.p.value < 0.05/n_tests,
        sign_early  = sign(estimate_early),
        sign_late   = sign(estimate_late),
        size_inc    = abs(estimate_early) < abs(estimate_late),
        change_sign = sign_early != sign_late,
        Change      = case_when(
                        ! significant ~ "Insignificant",
                        change_sign   ~ "Change sign",
                        size_inc      ~ "Size increasing",
                        ! size_inc    ~ "Size decreasing"
                      )
      ) %>% 
      select(score_name, children, Change, significant, change_sign, 
        sign_early, sign_late) %>% 
      mutate(children = ifelse(children, "Respondents", "Parents")) %>% 
      pivot_wider(names_from = children, 
        values_from = c(Change, significant, change_sign, sign_early, 
        sign_late)) %>% 
      mutate(overall = case_when(
              ! change_sign_Parents & ! change_sign_Respondents & 
                sign_late_Parents == sign_early_Parents ~ 
                "Consistent",
              TRUE ~ "Changes direction"
            ))


stopifnot(sum(summary_period$Change_Parents == "Insignificant") == 33)
stopifnot(sum(summary_period$Change_Respondents == "Size increasing") == 4)
stopifnot(sum(summary_period$Change_Respondents == "Change sign") == 1)
stopifnot(sum(summary_period$Change_Respondents == "Insignificant") == 28)
```

As an alternative approach, we run regressions interacting polygenic scores with
birth year, median split at 1950 ("early born" versus "late born"). We use both
*number of children* and *number of siblings* as a dependent variable. We weight
using age/qualification cells, and further adjust for selection in the parents'
generation (see above).

Tables \@ref(tab:tbl-siblings-by-period) and
\@ref(tab:tbl-children-by-period) summarize the results. There is no
evidence for changes in selection effects within the parents' generation. In the
respondents' generation, effect sizes were significantly larger in absolute size 
among the later-born for four polygenic scores: cognitive ability, EA2, EA3 and
extraversion. These changes are inconsistent with the intergenerational change, 
where estimated effect sizes were larger among the earlier, parents' generation.
One score, conscientiousness, showed a significant change in sign, from 
negative to positive effects on fertility.

Overall, there is weak evidence for change over time. The clearest results are
that (a) the direction of selection, and (b) the pattern of relative effect sizes
across scores, are broadly consistent over time.


```{r tbl-siblings-by-period}

period_table <- function (tbl, caption) {
  tbl %>% 
      as_hux() %>%
      set_bold(1, everywhere) %>% 
      set_bottom_border(1, everywhere) %>% 
      
      add_footnote(glue("Significance is measured at p < 0.05/{n_tests}"), 
        border = 0.4, font_size = 8) %>% 
      set_contents(1, everywhere, 
        c("Change", "Number of scores")) %>% 
      set_right_border(final(1), 1, 0) %>% 
      set_top_padding(1) %>% 
      set_bottom_padding(1) %>%
      set_left_padding(everywhere, 1, 0) %>% 
      set_right_padding(everywhere, 2, 0) %>% 
      set_caption(caption) %>% 
      set_caption_width(1)
}

summary_period %>% 
      group_by(Change_Parents) %>% 
      count() %>% 
      period_table("Change in selection effects between parents of early and late born respondents (regressions on number of siblings).")
      
```

```{r tbl-children-by-period}
summary_period %>% 
      group_by(Change_Respondents) %>% 
      count() %>% 
      period_table("Change in selection effects between early and late born respondents (regressions on number of children).")
```

Figure \@ref(fig:plot-siblings-townsend) plots effects on *number of siblings* by
Townsend deprivation quintile of birth area.

```{r plot-siblings-townsend, fig.cap = "Selection effects in the parents' generation by Townsend deprivation quintile of birth area.", fig.align = "center"}

drake::loadd(res_townsend_parents)
drake::loadd(famhist_townsend_71)
n_regs <- nrow(res_townsend_parents)

res_townsend_parents %>% 
  mutate(
    `Townsend quintile` = sub("Quin71(\\d):.*", "\\1", term),
    `Townsend quintile` = add_n(`Townsend quintile`, "Quin71", dv = "n_sibs", 
                                  data = famhist_townsend_71)
  ) %>% 
  standard_ggplot(fill_col    = `Townsend quintile`,
                    n_regs    = n_regs, 
                    score_col = score_name, 
                    order_idx = 5
                  )

rm(famhist_townsend_71)
```



\clearpage
\FloatBarrier
## Selection effects on raw polygenic scores


```{r calc-regs-controlled}

drake::loadd(res_all)

res_all_controlled <- res_all %>% 
  select(reg.type, dep.var, term, estimate) %>% 
  pivot_wider(names_from = c(reg.type, dep.var), values_from = estimate) %>% 
  mutate(
    consistent = sign(`raw_N siblings`) == sign(`raw_N children`) &
                 sign(`raw_N siblings`) == sign(`controlled_N siblings`) &
                 sign(`raw_N siblings`) == sign(`controlled_N children`)
  )

n_pgs  <- as.double(nrow(res_all_controlled))
prop_consistent_controlled <- mean(res_all_controlled$consistent)
controlled_smaller_sibs <- abs(res_all_controlled[["controlled_N siblings"]]) < 
  abs(res_all_controlled[["raw_N siblings"]])
controlled_smaller_children <- abs(res_all_controlled[["controlled_N children"]]) < 
  abs(res_all_controlled[["raw_N children"]])
median_prop_sibs <- median(abs(res_all_controlled[["controlled_N siblings"]]) /
    abs(res_all_controlled[["raw_N siblings"]]))
median_prop_children <- median(abs(res_all_controlled[["controlled_N children"]]) /
    abs(res_all_controlled[["raw_N children"]]))

```

Figure \@ref(fig:plot-regs-controlled-pcs) compares selection effects on
polygenic scores residualized for the top 100 principal components of the
genetic data, to selection effects on raw, unresidualized polygenic scores. In
siblings regressions, effect sizes are larger for raw scores  -- sometimes much
larger, as in the case of height. `r sum(controlled_smaller_sibs)` out of 
`r n_pgs` "raw" effect sizes have a larger absolute value than the corresponding
"residualized" effect size. The median proportion between raw and controlled
effect sizes is `r median_prop_sibs`. Among the children regressions, this no
longer holds. Effect sizes are barely affected by controlling for principal
components.

Overall, `r prop_consistent_controlled * 100` per cent of effect sizes
are consistently signed across all four regressions (on children and
siblings, and with and without residualization).

```{r plot-regs-controlled-pcs, fig.cap = "Selection effects using unresidualized polygenic scores on number of siblings/children."}

drake::loadd(res_all)

n_regs <- as.double(nrow(res_all))
res_all %>% 
      mutate(
        term     = pretty_names(term),
        term     = reorder_within(term, estimate, dep.var, order_abs(1)),
        dep.var  = fct_inorder(dep.var),
        reg.type = fct_recode(reg.type, 
                               `Resid. for 100 PCs`  = "controlled", 
                               Unresidualized        = "raw"
                             )
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(y = term, x = estimate, fill = reg.type, 
        colour = p < 0.05/{{n_regs}})) +
        my_vline + 
        facet_wrap(vars(dep.var), scales = "free") +
        pval_geom_point(position = position_dodge(0.4), alpha = 0.8) +
        my_labs() +
        pval_scale() +
        my_fill_scale(n = 2, guide = guide_legend(title = "")) +
        my_theme() +
        scale_y_reordered() +
        theme(
          axis.text.x  = element_text(size = 7),
          legend.text  = element_text(size = 8),
          legend.title = element_text(size = 9)
        )

```

```{r}
drake::loadd(res_pcs)
n_regs <- as.double(nrow(res_pcs))
prop_sig <- res_pcs %>% 
      group_by(dep.var) %>% 
      summarize(sig = sum(p.value <= 0.05/nrow(res_pcs))) %>% 
      pull(sig, name = dep.var)
```

To get a further insight into this we regress *n siblings* and *n children*
on individual principal components. As Figure \@ref(fig:plot-pc-regs)
shows, effects are larger and more significant in siblings regressions.
`r prop_sig[["N siblings"]]` principal components significantly
predicted number of siblings, while only `r prop_sig[["N children"]]`
significantly predicted number of children.

```{r plot-pc-regs, fig.cap = "Selection effects of 100 principal components of genetic data. Absolute effect sizes are plotted. Each dot represents one bivariate regression. Points are jittered on the Y axis.", fig.height = 2}

res_pcs %>% 
      mutate(
        estimate = abs(estimate),
        dep.var  = fct_relevel(dep.var, "N children")
      ) %>% 
      rename(p = p.value) %>% 
      ggplot(aes(estimate, dep.var, colour = p < 0.05/{{n_regs}})) + 
        geom_jitter(height = 0.15, width = 0, alpha = 0.8) + 
        scale_x_log10(labels = scales::label_number(accuracy = 0.0001)) +
        labs(x = "Effect size (log scale)", y = "") + 
        pval_scale(aesthetics = "colour")
        

```


\clearpage
\FloatBarrier
## Selection controlling for age at first live birth: respondents' parents


```{r calc-age-birth-parents}

drake::loadd(res_age_birth_parents)
drake::loadd(res_all)

res_sibs_unc <- filter(res_all, dep.var == "N siblings", reg.type == "controlled")
compare_abp_unc <- res_age_birth_parents %>% 
      filter(! term %in% c("fath_age_birth", "moth_age_birth")) %>% 
      left_join(res_sibs_unc, by = "term")

cor_unc_fath <- with(
  compare_abp_unc %>% filter(control == "fath_age_birth"), 
  cor(estimate.x, estimate.y))
cor_unc_moth <- with(
  compare_abp_unc %>% filter(control == "moth_age_birth"), 
  cor(estimate.x, estimate.y))
```

Among the parents' generation, we can control for age at first live
birth using the subsets of respondents who reported their mother's or
father's age, and who had no elder siblings. We run regressions on
*number of siblings* on these subsets, controlling for either parent's
age at their birth. Figure \@ref(fig:plot-age-birth-parents) shows the
results. Effect sizes are very similar, whether controlling for father's
or mother's age. As in the respondents' generation, effect sizes
are negatively correlated with the effect sizes from bivariate
regressions without the age at birth control (father's age at birth: 
$\rho$ `r cor_unc_fath`; mother's age at birth: $\rho$ `r cor_unc_moth`).

```{r plot-age-birth-parents, fig.cap = "Selection effects (parents' generation) among eldest siblings, controlling for parents' age at birth."}


n_regs <- as.double(nrow(res_age_birth_parents)/2)

res_age_birth_parents %>% 
    filter(! term %in% c("fath_age_birth", "moth_age_birth")) %>% 
    mutate(
      Control = ifelse(control == "fath_age_birth", 
                  "Father's age at birth", 
                  "Mother's age at birth"
                )
    ) %>% 
    standard_ggplot(fill_col = Control, n_regs = n_regs, order_idx = 2)
    

```


\clearpage
\FloatBarrier
## Effects of polygenic scores on age at first live birth


```{r calc-age-flb-dv}

drake::loadd(res_age_flb_dv)
drake::loadd(res_age_birth_parents_dv)

compare_flb_dv <- left_join(
        res_age_flb_dv           %>% select(term, estimate),
        res_age_birth_parents_dv %>% select(term, estimate, dep.var),
        by = "term"
      )

cor_flb_fath <- with(compare_flb_dv %>% filter(dep.var == "fath_age_birth"),
        cor(estimate.x, estimate.y)
      )
cor_flb_moth <- with(compare_flb_dv %>% filter(dep.var == "moth_age_birth"),
        cor(estimate.x, estimate.y)
      )
```

Our results suggest that polygenic scores may directly correlate with age at
first live birth. Figure \@ref(fig:plot-age-flb-dv) plots estimated effect sizes
from bivariate regressions for respondents. Figure
\@ref(fig:plot-age-birth-parents-dv) does the same for their parents, using only
eldest siblings.[^only-eldest] Effect sizes are reasonably large. They are also
highly correlated across generations. Effect sizes of polygenic scores on
father's age at own birth, and on own age at first live birth, have a
correlation of `r cor_flb_fath`; for mother's age and own age it is 
`r cor_flb_moth`.

[^only-eldest]: Parental AFLB can only be calculated for this group.

```{r plot-age-flb-dv, fig.cap = "Effects of polygenic scores on age at first live birth."}

n_regs <- as.double(nrow(res_age_flb_dv))

res_age_flb_dv %>% standard_ggplot(n_regs = n_regs)

```


```{r plot-age-birth-parents-dv, fig.cap = "Effects of polygenic scores on parents' age at respondent's birth, eldest siblings."}

n_regs <- as.double(nrow(res_age_birth_parents_dv))

res_age_birth_parents_dv %>% 
      mutate(
        "Dependent variable" = dplyr::recode(dep.var, 
          "fath_age_birth" = "Father's AFLB", 
          "moth_age_birth" = "Mother's AFLB"
        )
      ) %>% 
      standard_ggplot(fill_col = `Dependent variable`, n_regs = n_regs)

```


\FloatBarrier
\clearpage
## Controlling for earnings and education


```{r plot-res-educ, fig.cap = "Selection effects controlling for education (left education before 16, 16-18, or after 18). Raw effects are shown for comparison."}

drake::loadd(res_ee_control)
drake::loadd(res_all)

res_ee_control %<>% filter(term == score_name)


res_all %<>% 
  filter(term == score_name, reg.type == "controlled", dep.var == "N children") %>% 
  select(-dep.var, -reg.type)

res_compare <- bind_rows(
                 Raw = res_all, 
                 Controlled = res_ee_control, 
                 .id = "Regression"
               )

standard_ggplot(res_compare, fill_col = Regression, n_regs = 33)
  
```



\FloatBarrier
\clearpage
## Within-siblings regressions


```{r plot-res-fe-fertility, fig.cap = "Selection effects controlling for sibling-group fixed effects, with and without a control for education (left education before 16, 16-18, or after 18)."}

drake::loadd(res_fe_fertility)

standard_ggplot(res_fe_fertility, fill_col = Regression, n_regs = 33, fill_direction = -1)
  
```

\FloatBarrier
\clearpage
## Genetic correlations with EA3


```{r plot-rgs-by-effect-size, fig.cap = "Selection effects plotted against genetic correlation with EA3."}

# TODO: ref source of data.

drake::loadd(rgs)
drake::loadd(res_all)

rgs <- left_join(rgs, 
        res_all %>% filter(reg.type == "controlled", dep.var == "N children"), 
        by = c("p2" = "term"))

rgs_cor <- cor.test(~ rg + estimate, 
                      data = rgs %>%
                               filter(
                                 p2 != "EA3_excl_23andMe_UK", 
                                 p2 != "EA2_noUKB"
                               )
                    )
rgs %>% 
      filter(p2 != "EA3_excl_23andMe_UK", p2 != "EA2_noUKB") %>% 
      ggplot(aes(rg, estimate)) + 
        geom_point() +
        my_vline +
        my_hline +
        labs(x = "Genetic correlation with EA3", y = "Effect on number of children")
          
```

Another way to examine the "earnings" theory of natural selection is to compare
selection effects of polygenic scores with their genetic correlation with educational
attainment (EA3). Since EA3 strongly predicts earnings, if earnings drives
differences in fertility, we'd expect a correlation between the two sets of
results. Figure \@ref(fig:plot-rgs-by-effect-size) shows this is so: the 
correlation, after excluding EA2, is `r rgs_cor$estimate`. Genetic
correlations were calculated using LD score regression from GWAS summary
statistics.


\clearpage
\FloatBarrier
## Effects on inequality

Table \@ref(tab:tbl-inequality-effects) estimates differences in children's mean
polygenic scores between the highest and lowest income groups. Column "With
selection" uses respondents' scores weighted by Age/Qualification times number
of children. Column "Without selection" uses scores weighted by
Age/Qualification only, i.e. if all couples had the same number of children.


```{r tbl-inequality-effects}

ineq_comp %>% 
      select(
        PGS                 = score, 
        `Without selection` = diff_hypoth,
        `With selection`    = diff_children, 
        `% change`          = pct_change
      ) %>% 
      mutate(
        PGS = pretty_names(PGS)
      ) %>% 
      arrange(desc(`% change`)) %>% 
      as_hux() %>% 
      set_align(everywhere, -1, "right") %>% 
      set_number_format(everywhere, 2:3, 3) %>% 
      set_number_format(everywhere, 4, 1) %>% 
      set_caption("Differences in polygenic scores between highest and lowest income group.") %>% 
      theme_compact() %>% 
      set_width(0.75) %>% 
      set_col_width(c(.4, .2, .2, .2)) %>% 
      set_bottom_border(final(1), everywhere) %>% 
      set_position("center")
```


\clearpage
\FloatBarrier

# Proofs

```{r include-model-appendix, child="model-appendix.Rmd"}

```

\clearpage
\FloatBarrier
# References
